<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>transposonmapper.plotting API documentation</title>
<meta name="description" content="" />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>transposonmapper.plotting</code></h1>
</header>
<section id="section-intro">
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">from .dna_features_plot import dna_features_plot
from .profile_genome_plot import profile_genome_plot

__all__ = [
    &#34;dna_features_plot&#34;,
    &#34;profile_genome_plot&#34;
]</code></pre>
</details>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="transposonmapper.plotting.dna_features_plot"><code class="name flex">
<span>def <span class="ident">dna_features_plot</span></span>(<span>chrom, dna_df2, roi_start, roi_end, len_chr, plotting, variable)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def dna_features_plot(chrom, dna_df2,roi_start,roi_end,len_chr,plotting,variable):    


    if plotting == True:
        create_plottitle = chrom
        noncoding_color = &#34;#002538&#34;
        essential_color = &#34;#10e372&#34;
        nonessential_color = &#34;#d9252e&#34;
        codingdna_color = &#39;#29a7e6&#39;
        textcolor = &#34;#000000&#34;

        textsize = 14

    plt.figure(figsize=(19,9))
    grid = plt.GridSpec(20, 1, wspace=0.0, hspace=0.01)

    axc = plt.subplot(grid[19,0])

    axc.tick_params(labelsize=textsize)
    axc.set_yticklabels([])
    axc.tick_params(
        axis=&#39;x&#39;,          # changes apply to the x-axis
        which=&#39;both&#39;,      # both major and minor ticks are affected
        bottom=False,      # ticks along the bottom edge are off
        top=False,         # ticks along the top edge are off
        labelbottom=False) # labels along the bottom edge are off

    axc.tick_params(
        axis=&#39;y&#39;,          # changes apply to the y-axis
        which=&#39;both&#39;,      # both major and minor ticks are affected
        left=False,        # ticks along the bottom edge are off
        right=False,       # ticks along the top edge are off
        labelleft=False)   # labels along the bottom edge are off


    ax = plt.subplot(grid[0:19,0])
    

    ax.grid(linestyle=&#39;-&#39;, alpha=1.0)
    ax.tick_params(labelsize=textsize)

    ax.tick_params(axis=&#39;x&#39;, which=&#39;major&#39;, pad=30)
    ax.ticklabel_format(axis=&#39;x&#39;, style=&#39;sci&#39;, scilimits=(0,0))
    ax.xaxis.get_offset_text().set_fontsize(textsize)
    ax.set_xlabel(&#34;Basepair position on chromosome &#34;+chrom, fontsize=textsize, color=textcolor, labelpad=10)
    ax.set_title(create_plottitle, fontsize=textsize, color=textcolor)


    feature_middle_pos_list = []
    sum_bp = 0
    for x in dna_df2[&#39;Nbasepairs&#39;]:
        feature_middle_pos_list.append(x/2 + sum_bp)
        sum_bp += x
    
    feature_width_list = list(dna_df2[&#39;Nbasepairs&#39;])


    barcolor_list = []
    for feature in dna_df2[&#39;Feature_name&#39;]:
        if feature == &#39;noncoding&#39;:
            barcolor_list.append(noncoding_color)
        elif dna_df2.loc[dna_df2[&#39;Feature_name&#39;] == feature][&#39;Essentiality&#39;].iloc[0] == False:
            barcolor_list.append(nonessential_color)
        elif dna_df2.loc[dna_df2[&#39;Feature_name&#39;] == feature][&#39;Essentiality&#39;].iloc[0] == True:
            barcolor_list.append(essential_color)
        elif dna_df2.loc[dna_df2[&#39;Feature_name&#39;] == feature][&#39;Essentiality&#39;].iloc[0] == None:
            barcolor_list.append(codingdna_color)

    

    legend_noncoding = mpatches.Patch(color=noncoding_color, label=&#34;Noncoding DNA&#34;)
    legend_essential = mpatches.Patch(color=essential_color, label=&#34;Annotated essential genes&#34;)
    legend_nonessential = mpatches.Patch(color=nonessential_color, label=&#34;Nonessential genes&#34;)
    legend_coding = mpatches.Patch(color=codingdna_color, label=&#34;Other genomic regions&#34;)
    leg = ax.legend(handles=[legend_noncoding, legend_essential, legend_nonessential, legend_coding]) #ADD

    for text in leg.get_texts():
        text.set_color(textcolor)
    
    if variable == &#34;insertions&#34;:
        ax.bar(feature_middle_pos_list, list(dna_df2[&#39;Ninsertions&#39;]), feature_width_list, color=barcolor_list)
        ax.set_ylabel(&#34;Transposons per region&#34;, fontsize=textsize, color=textcolor)
    elif variable == &#34;reads&#34;:
        ax.bar(feature_middle_pos_list, list(dna_df2[&#39;Nreads&#39;]), feature_width_list, color=barcolor_list)
        ax.set_ylabel(&#34;Reads per region&#34;, fontsize=textsize, color=textcolor)


    if roi_start != None and roi_end != None and roi_start &lt; len_chr and roi_end &lt; len_chr:
        ax.set_xlim(roi_start, roi_end)
    else:
        ax.set_xlim(0, len_chr)



    

    l = 0
    counter = 0
    for width in feature_width_list:
        if dna_df2.loc[counter][4] == True:
            axc.axvspan(l,l+width,facecolor=essential_color,alpha=0.3)
        elif dna_df2.loc[counter][4] == False and not dna_df2.loc[counter][0] == &#39;noncoding&#39;:
            axc.axvspan(l,l+width,facecolor=nonessential_color,alpha=0.3)
        elif dna_df2.loc[counter][4] == None and not dna_df2.loc[counter][0] == &#39;noncoding&#39;:
            axc.axvspan(l,l+width,facecolor=codingdna_color,alpha=0.5)
        l += width
        counter += 1
    if roi_start != None and roi_end != None and roi_start &lt; len_chr and roi_end &lt; len_chr:
        axc.set_xlim(roi_start, roi_end)
    else:
        axc.set_xlim(0, len_chr)</code></pre>
</details>
</dd>
<dt id="transposonmapper.plotting.profile_genome_plot"><code class="name flex">
<span>def <span class="ident">profile_genome_plot</span></span>(<span>bar_width, l_genome, allinsertionsites_list, allcounts_binnedlist, summed_chr_length_dict, middle_chr_position, chrom_list, variable, genes_currentchrom_pos_list, gene_pos_dict)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def profile_genome_plot(bar_width,l_genome,allinsertionsites_list,allcounts_binnedlist,summed_chr_length_dict,
                         middle_chr_position,chrom_list,variable,genes_currentchrom_pos_list,gene_pos_dict):

    _,essential_file,_=load_default_files()
    genes_essential_list=list_known_essentials(essential_file)


    plt.figure(figsize=(19.0,9.0))
    grid = plt.GridSpec(20, 1, wspace=0.0, hspace=0.0)

    textsize = 12
    textcolor = &#34;#000000&#34;
    barcolor= &#34;#333333&#34;
    chrom_color=(0.9,0.9,0.9,1.0)
    essential_face_color=&#34;#00F28E&#34;
    non_essential_face_color=&#34;#F20064&#34;
    alpha=0.8
    
    binsize = bar_width
    ax = plt.subplot(grid[0:19,0])
    ax.grid(False)
    ax.tick_params(axis=&#39;x&#39;, which=&#39;major&#39;, pad=30)
    
    
    
    axc = plt.subplot(grid[19,0])
    
    axc.set_xlim(0,l_genome)
    axc.tick_params(
        axis=&#39;x&#39;,          # changes apply to the x-axis
        which=&#39;both&#39;,      # both major and minor ticks are affected
        bottom=False,      # ticks along the bottom edge are off
        top=False,         # ticks along the top edge are off
        labelbottom=False) # labels along the bottom edge are off

    axc.tick_params(
        axis=&#39;y&#39;,          # changes apply to the y-axis
        which=&#39;both&#39;,      # both major and minor ticks are affected
        left=False,        # ticks along the bottom edge are off
        right=False,       # ticks along the top edge are off
        labelleft=False)   # labels along the bottom edge are off
    
    ax.set_xlim(0,l_genome)
    
    # bar lines
    ax.bar(allinsertionsites_list,allcounts_binnedlist,width=binsize,color=barcolor)
   
    
    # chromosome lines
    for chrom in summed_chr_length_dict:
        ax.axvline(x = summed_chr_length_dict.get(chrom), linestyle=&#39;-&#39;, color=chrom_color)

    ax.set_xticks(middle_chr_position)
    ax.set_xticklabels(chrom_list, fontsize=textsize)
    
    # Axis labels
    if variable == &#34;transposons&#34;:
        ax.set_ylabel(&#39;Transposon Count&#39;, fontsize=textsize, color=textcolor)
    elif variable == &#34;reads&#34;:
        ax.set_ylabel(&#39;Read Count&#39;, fontsize=textsize, color=textcolor)

    # colored bars in the bottom 
    for gene in genes_currentchrom_pos_list:
        if not gene_pos_dict.get(gene)[0] == &#39;Mito&#39;:
            gene_start_pos = summed_chr_length_dict.get(gene_pos_dict.get(gene)[0]) + int(gene_pos_dict.get(gene)[1])
            gene_end_pos = summed_chr_length_dict.get(gene_pos_dict.get(gene)[0]) + int(gene_pos_dict.get(gene)[2])
            if gene in genes_essential_list:
                axc.axvspan(gene_start_pos,gene_end_pos,facecolor=essential_face_color,alpha=alpha)
            else:
                axc.axvspan(gene_start_pos,gene_end_pos,facecolor=non_essential_face_color,alpha=alpha)</code></pre>
</details>
</dd>
</dl>
</section>
<section>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="transposonmapper" href="../index.html">transposonmapper</a></code></li>
</ul>
</li>
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="">
<li><code><a title="transposonmapper.plotting.dna_features_plot" href="#transposonmapper.plotting.dna_features_plot">dna_features_plot</a></code></li>
<li><code><a title="transposonmapper.plotting.profile_genome_plot" href="#transposonmapper.plotting.profile_genome_plot">profile_genome_plot</a></code></li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html>