<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>transposonmapper.processing.genomicfeatures_dataframe API documentation</title>
<meta name="description" content="" />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>transposonmapper.processing.genomicfeatures_dataframe</code></h1>
</header>
<section id="section-intro">
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">import os, sys
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib.patches as mpatches
import numpy as np 
import pkg_resources


from transposonmapper.properties import chromosome_position,gene_aliases
from transposonmapper.processing import chromosome_name_wigfile

from transposonmapper.processing.read_sgdfeatures import sgd_features


from transposonmapper.importing import (
    load_default_files, load_sgd_tab
)


from transposonmapper.processing.dna_features_helpers import (build_dataframe, input_region, intergenic_regions, read_pergene_file, 
                                                              read_wig_file,gene_location,checking_features)
from transposonmapper.plotting.dna_features_plot import dna_features_plot

def dna_features(region, wig_file, pergene_insertions_file, variable=&#34;reads&#34;, plotting=True, savefigure=False, verbose=True):
    &#34;&#34;&#34;This scripts takes a user defined genomic region (i.e. chromosome number, region or gene) and creates a dataframe including information about all genomic features in the chromosome (i.e. genes, nc-DNA etc.).
    This can be used to determine the number of reads outside the genes to use this for normalization of the number of reads in the genes.
    Output is a dataframe including major information about all genomic features and optionally a barplot indicating the number of transposons per genomic region.
    A genomic region is here defined as a gene (separated as annotated essential and not essential), telomere, centromere, ars etc.
    This can be used for identifying neutral regions (i.e. genomic regions that, if inhibited, do not influence the fitness of the cells).
    This function can be used for normalizing the transposon insertions per gene using the neutral regions.
    
              
    
    Parameters
    ----------
    region : str
        - Region: e.g. chromosome number (either a normal number between 1 and 16 or in roman numerals between I and XVI), a list like [&#39;V&#39;, 0, 14790] which creates a barplot between basepair 0 and 14790) or a genename.

    wig_file : str
        absolute path for the wig file location
    pergene_insertions_file : str 
        asbsoulte path for the _pergene_insertions.txt file location 
    variable : str, optional
        By default &#34;reads&#34;. It could be &#34;transposons&#34;or &#34;reads&#34;. This would be used for the plotting if True 
    plotting : bool, optional
        Whether or not producing a bar plot with the reads/insertions per genomic location in the region, by default True
    savefigure : bool, optional
        Whether or not saving the plot in the same folder as the datafiles, by default False
    verbose : bool, optional
        Determines how much textual feedback is given. When set to False, only warnings will be shown. By default True

    Outputs
    -------------------
    - dna_df2: Dataframe containing information about the selected chromosome. This includes the following columns:

        Notes
    ------------------

    __Date__: 14-10-2020
    __Author__: Gregory van Beek
    &#34;&#34;&#34;

    # If necessary, load default files
    gff_file, essentials_file, gene_information_file = load_default_files()
    sgd_features_file=load_sgd_tab()

    # Verify presence of files
    data_files = {
        &#34;gff3&#34;: gff_file,
        &#34;essentials&#34;: essentials_file,
        &#34;gene_names&#34;: gene_information_file,
        &#34;sgd_features&#34;: sgd_features_file
    }

    for filetype, file_path in data_files.items():
        assert file_path, f&#34;{filetype} not found at {file_path}&#34;


    variable = variable.lower()
    if plotting == True:
        create_plottitle = &#39;&#39;

# DETERMINE INPUTTED REGION

    roi_start,roi_end,region_type,chrom=input_region(region=region,verbose=verbose)

    

#READ WIG FILE FOR GETTING LOCATIONS OF ALL TN INSERTIONS

    insrt_in_chrom_list,reads_in_chrom_list=read_wig_file(wig_file=wig_file,chrom=chrom)


# READ PERGENE_INSERTIONS FILE FOR LOCATION OF ALL INSERTIONS PER EACH GENE.

    gene_position_dict=read_pergene_file(pergene_insertions_file=pergene_insertions_file,chrom=chrom)

# DETERMINE THE LOCATION GENOMIC FEATURES IN THE CURRENT CHROMOSOME AND STORE THIS IN A DICTIONARY

    dna_dict,start_chr,end_chr,len_chr,feature_orf_dict=gene_location(chrom,gene_position_dict,verbose)

## GET FEATURES FROM INTERGENIC REGIONS 

    dna_dict,genomicregions_list=intergenic_regions(chrom,start_chr,dna_dict)


    ### TEST IF ELEMENTS IN FEATURE_ORF_DICT FOR SELECTED CHROMOSOME ARE THE SAME AS THE GENES IN GENE_POSITION_DICT BY CREATING THE DICTIONARY FEATURE_POSITION_DICT CONTAINING ALL THE GENES IN FEATURE_ORF_DICT WITH THEIR CORRESPONDING POSITION IN THE CHROMOSOME
    checking_features(feature_orf_dict,chrom,gene_position_dict,verbose)

    dna_df2=build_dataframe(dna_dict,start_chr,end_chr,insrt_in_chrom_list,reads_in_chrom_list,genomicregions_list,chrom)

    #PRINT INFORMATION FOR THE SELECTED GENE
    if region_type == &#39;Gene&#39;:
        for region_info in dna_df2.itertuples():
            if region_info.Feature_name == region.upper() or region_info.Standard_name == region.upper():
                print(region_info)

    
    # CREATE BAR PLOT 
    dna_features_plot(chrom, dna_df2,roi_start,roi_end,len_chr,plotting,variable)

    if savefigure == True:
        file_dirname=pkg_resources.resource_filename(&#34;transposonmapper&#34;, &#34;data_files/&#34;)
        if variable == &#39;reads&#39;:
            saving_name = os.path.join(file_dirname,&#39;GenomicFeaturesReads_Barplot_Chrom&#39;+chrom+&#39;_NonNormalized&#39;)
        else:
            saving_name = os.path.join(file_dirname,&#39;GenomicFeaturesInsertions_Barplot_Chrom&#39;+chrom+&#39;_NonNormalized&#39;)
        plt.savefig(saving_name, orientation=&#39;landscape&#39;, dpi=200)
        plt.close()

    return(dna_df2)</code></pre>
</details>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="transposonmapper.processing.genomicfeatures_dataframe.dna_features"><code class="name flex">
<span>def <span class="ident">dna_features</span></span>(<span>region, wig_file, pergene_insertions_file, variable='reads', plotting=True, savefigure=False, verbose=True)</span>
</code></dt>
<dd>
<div class="desc"><p>This scripts takes a user defined genomic region (i.e. chromosome number, region or gene) and creates a dataframe including information about all genomic features in the chromosome (i.e. genes, nc-DNA etc.).
This can be used to determine the number of reads outside the genes to use this for normalization of the number of reads in the genes.
Output is a dataframe including major information about all genomic features and optionally a barplot indicating the number of transposons per genomic region.
A genomic region is here defined as a gene (separated as annotated essential and not essential), telomere, centromere, ars etc.
This can be used for identifying neutral regions (i.e. genomic regions that, if inhibited, do not influence the fitness of the cells).
This function can be used for normalizing the transposon insertions per gene using the neutral regions.</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>region</code></strong> :&ensp;<code>str</code></dt>
<dd>
<ul>
<li>Region: e.g. chromosome number (either a normal number between 1 and 16 or in roman numerals between I and XVI), a list like ['V', 0, 14790] which creates a barplot between basepair 0 and 14790) or a genename.</li>
</ul>
</dd>
<dt><strong><code>wig_file</code></strong> :&ensp;<code>str</code></dt>
<dd>absolute path for the wig file location</dd>
<dt><strong><code>pergene_insertions_file</code></strong> :&ensp;<code>str </code></dt>
<dd>asbsoulte path for the _pergene_insertions.txt file location</dd>
<dt><strong><code>variable</code></strong> :&ensp;<code>str</code>, optional</dt>
<dd>By default "reads". It could be "transposons"or "reads". This would be used for the plotting if True</dd>
<dt><strong><code>plotting</code></strong> :&ensp;<code>bool</code>, optional</dt>
<dd>Whether or not producing a bar plot with the reads/insertions per genomic location in the region, by default True</dd>
<dt><strong><code>savefigure</code></strong> :&ensp;<code>bool</code>, optional</dt>
<dd>Whether or not saving the plot in the same folder as the datafiles, by default False</dd>
<dt><strong><code>verbose</code></strong> :&ensp;<code>bool</code>, optional</dt>
<dd>Determines how much textual feedback is given. When set to False, only warnings will be shown. By default True</dd>
</dl>
<h2 id="outputs">Outputs</h2>
<ul>
<li>
<p>dna_df2: Dataframe containing information about the selected chromosome. This includes the following columns:</p>
<h2 id="notes">Notes</h2>
</li>
</ul>
<p><strong>Date</strong>: 14-10-2020
<strong>Author</strong>: Gregory van Beek</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def dna_features(region, wig_file, pergene_insertions_file, variable=&#34;reads&#34;, plotting=True, savefigure=False, verbose=True):
    &#34;&#34;&#34;This scripts takes a user defined genomic region (i.e. chromosome number, region or gene) and creates a dataframe including information about all genomic features in the chromosome (i.e. genes, nc-DNA etc.).
    This can be used to determine the number of reads outside the genes to use this for normalization of the number of reads in the genes.
    Output is a dataframe including major information about all genomic features and optionally a barplot indicating the number of transposons per genomic region.
    A genomic region is here defined as a gene (separated as annotated essential and not essential), telomere, centromere, ars etc.
    This can be used for identifying neutral regions (i.e. genomic regions that, if inhibited, do not influence the fitness of the cells).
    This function can be used for normalizing the transposon insertions per gene using the neutral regions.
    
              
    
    Parameters
    ----------
    region : str
        - Region: e.g. chromosome number (either a normal number between 1 and 16 or in roman numerals between I and XVI), a list like [&#39;V&#39;, 0, 14790] which creates a barplot between basepair 0 and 14790) or a genename.

    wig_file : str
        absolute path for the wig file location
    pergene_insertions_file : str 
        asbsoulte path for the _pergene_insertions.txt file location 
    variable : str, optional
        By default &#34;reads&#34;. It could be &#34;transposons&#34;or &#34;reads&#34;. This would be used for the plotting if True 
    plotting : bool, optional
        Whether or not producing a bar plot with the reads/insertions per genomic location in the region, by default True
    savefigure : bool, optional
        Whether or not saving the plot in the same folder as the datafiles, by default False
    verbose : bool, optional
        Determines how much textual feedback is given. When set to False, only warnings will be shown. By default True

    Outputs
    -------------------
    - dna_df2: Dataframe containing information about the selected chromosome. This includes the following columns:

        Notes
    ------------------

    __Date__: 14-10-2020
    __Author__: Gregory van Beek
    &#34;&#34;&#34;

    # If necessary, load default files
    gff_file, essentials_file, gene_information_file = load_default_files()
    sgd_features_file=load_sgd_tab()

    # Verify presence of files
    data_files = {
        &#34;gff3&#34;: gff_file,
        &#34;essentials&#34;: essentials_file,
        &#34;gene_names&#34;: gene_information_file,
        &#34;sgd_features&#34;: sgd_features_file
    }

    for filetype, file_path in data_files.items():
        assert file_path, f&#34;{filetype} not found at {file_path}&#34;


    variable = variable.lower()
    if plotting == True:
        create_plottitle = &#39;&#39;

# DETERMINE INPUTTED REGION

    roi_start,roi_end,region_type,chrom=input_region(region=region,verbose=verbose)

    

#READ WIG FILE FOR GETTING LOCATIONS OF ALL TN INSERTIONS

    insrt_in_chrom_list,reads_in_chrom_list=read_wig_file(wig_file=wig_file,chrom=chrom)


# READ PERGENE_INSERTIONS FILE FOR LOCATION OF ALL INSERTIONS PER EACH GENE.

    gene_position_dict=read_pergene_file(pergene_insertions_file=pergene_insertions_file,chrom=chrom)

# DETERMINE THE LOCATION GENOMIC FEATURES IN THE CURRENT CHROMOSOME AND STORE THIS IN A DICTIONARY

    dna_dict,start_chr,end_chr,len_chr,feature_orf_dict=gene_location(chrom,gene_position_dict,verbose)

## GET FEATURES FROM INTERGENIC REGIONS 

    dna_dict,genomicregions_list=intergenic_regions(chrom,start_chr,dna_dict)


    ### TEST IF ELEMENTS IN FEATURE_ORF_DICT FOR SELECTED CHROMOSOME ARE THE SAME AS THE GENES IN GENE_POSITION_DICT BY CREATING THE DICTIONARY FEATURE_POSITION_DICT CONTAINING ALL THE GENES IN FEATURE_ORF_DICT WITH THEIR CORRESPONDING POSITION IN THE CHROMOSOME
    checking_features(feature_orf_dict,chrom,gene_position_dict,verbose)

    dna_df2=build_dataframe(dna_dict,start_chr,end_chr,insrt_in_chrom_list,reads_in_chrom_list,genomicregions_list,chrom)

    #PRINT INFORMATION FOR THE SELECTED GENE
    if region_type == &#39;Gene&#39;:
        for region_info in dna_df2.itertuples():
            if region_info.Feature_name == region.upper() or region_info.Standard_name == region.upper():
                print(region_info)

    
    # CREATE BAR PLOT 
    dna_features_plot(chrom, dna_df2,roi_start,roi_end,len_chr,plotting,variable)

    if savefigure == True:
        file_dirname=pkg_resources.resource_filename(&#34;transposonmapper&#34;, &#34;data_files/&#34;)
        if variable == &#39;reads&#39;:
            saving_name = os.path.join(file_dirname,&#39;GenomicFeaturesReads_Barplot_Chrom&#39;+chrom+&#39;_NonNormalized&#39;)
        else:
            saving_name = os.path.join(file_dirname,&#39;GenomicFeaturesInsertions_Barplot_Chrom&#39;+chrom+&#39;_NonNormalized&#39;)
        plt.savefig(saving_name, orientation=&#39;landscape&#39;, dpi=200)
        plt.close()

    return(dna_df2)</code></pre>
</details>
</dd>
</dl>
</section>
<section>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="transposonmapper.processing" href="index.html">transposonmapper.processing</a></code></li>
</ul>
</li>
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="">
<li><code><a title="transposonmapper.processing.genomicfeatures_dataframe.dna_features" href="#transposonmapper.processing.genomicfeatures_dataframe.dna_features">dna_features</a></code></li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html>