<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>transposonmapper.processing.profileplot_genome_helpers API documentation</title>
<meta name="description" content="" />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>transposonmapper.processing.profileplot_genome_helpers</code></h1>
</header>
<section id="section-intro">
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">from transposonmapper.processing.chromosome_names_in_files import chromosome_name_bedfile
from transposonmapper.properties.get_chromosome_position import chromosome_position

import numpy as np 

def summed_chr(chr_length_dict):
    &#34;&#34;&#34;Create a dictionary where each value is the cumulative sum of all bp in each chromosomes

    Parameters
    ----------
    chr_length_dict : dict
        A dictionary describing the length of each chromosome. 

    Returns
    -------
    dict
        A dictionary where each value corresponds to the cumulative sum of the previous chromosomes lengths. 
    &#34;&#34;&#34;
    
    chrom_list = [&#39;I&#39;, &#39;II&#39;, &#39;III&#39;, &#39;IV&#39;, &#39;V&#39;, &#39;VI&#39;, &#39;VII&#39;, &#39;VIII&#39;, &#39;IX&#39;, &#39;X&#39;, &#39;XI&#39;, &#39;XII&#39;, &#39;XIII&#39;, &#39;XIV&#39;, &#39;XV&#39;, &#39;XVI&#39;]
    summed_chr_length_dict = {}
    summed_chr_length = 0
    for c in chrom_list:
        summed_chr_length_dict[c] = summed_chr_length
        summed_chr_length += chr_length_dict.get(c)    
     
    return summed_chr_length_dict


def length_genome(chr_length_dict):
    
    &#34;&#34;&#34;Output the length of the genome in bp 

    Parameters
    ----------

    chr_length_dict : dict
        A dictionary describing the length of each chromosome. 

    Returns
    -------
    int
        The length of the genome
    &#34;&#34;&#34;
    
    chrom_list = [&#39;I&#39;, &#39;II&#39;, &#39;III&#39;, &#39;IV&#39;, &#39;V&#39;, &#39;VI&#39;, &#39;VII&#39;, &#39;VIII&#39;, &#39;IX&#39;, &#39;X&#39;, &#39;XI&#39;, &#39;XII&#39;, &#39;XIII&#39;, &#39;XIV&#39;, &#39;XV&#39;, &#39;XVI&#39;]

    l_genome = 0
    for chrom in chrom_list:
        l_genome += int(chr_length_dict.get(chrom))
   
    
    return l_genome



def middle_chrom_pos(chr_length_dict):
    &#34;&#34;&#34;Defines the middle poit of each chromosome

    Parameters
    ----------
    chr_length_dict : dict
        A dictionary describing the length of each chromosome. 

    Returns
    -------
    list
        A list describing for each chromosome the middle point. 
    &#34;&#34;&#34;
    
    summed_chr_length_dict=summed_chr(chr_length_dict)
    
    l_genome=length_genome(chr_length_dict)
    
    
    middle_chr_position = []
    c1 = summed_chr_length_dict.get(&#39;I&#39;)
    for c in summed_chr_length_dict:
        if not c == &#39;I&#39;:
            c2 = summed_chr_length_dict.get(c)
            middle_chr_position.append(c1 + (c2 - c1)/2)
            c1 = c2
            
    c2 = l_genome
    middle_chr_position.append(c1 + (c2 - c1)/2)
    
    return middle_chr_position



def counts_genome(variable,bed_file,gff_file):
    &#34;&#34;&#34;Counts of reads or the transposons per chromosomes

    Parameters
    ----------
    variable : str
        &#34;transposons&#34; or &#34;reads&#34;
    bed_file : str
        absolute path of the location of the bedfile
    gff_file : str
        absolute path of the location of the gff file 

    Returns
    -------
    numpy.ndarray
        An array of the length of the genome with the counts of each variable per location in the genome. 
    &#34;&#34;&#34;
    
    with open(bed_file) as f:
        lines = f.readlines()
    
    chrom_names_dict, chrom_start_index_dict, chrom_end_index_dict= chromosome_name_bedfile(bed_file)
    chr_length_dict, chr_start_pos_dict, chr_end_pos_dict = chromosome_position(gff_file)
    
    summed_chr_length_dict=summed_chr(chr_length_dict)
    
    l_genome=length_genome(chr_length_dict)

    allcounts_list = np.zeros(l_genome)
    if variable == &#34;transposons&#34;:
        for line in lines[chrom_start_index_dict.get(&#34;I&#34;):chrom_end_index_dict.get(&#34;XVI&#34;)+1]:
            line = line.strip(&#39;\n&#39;).split()
            chrom_name = [k for k,v in chrom_names_dict.items() if v == line[0].replace(&#34;chr&#34;,&#39;&#39;)][0]
            allcounts_list[summed_chr_length_dict.get(chrom_name) + int(line[1])-1] += 1
    elif variable == &#34;reads&#34;:
        for line in lines[chrom_start_index_dict.get(&#34;I&#34;):chrom_end_index_dict.get(&#34;XVI&#34;)+1]:
            line = line.strip(&#39;\n&#39;).split()
            chrom_name = [k for k,v in chrom_names_dict.items() if v == line[0].replace(&#34;chr&#34;,&#39;&#39;)][0]
            allcounts_list[summed_chr_length_dict.get(chrom_name) + int(line[1])-1] += (int(line[4])-100)/20
    return allcounts_list

def binned_list(allcounts_list,bar_width):
    &#34;&#34;&#34;A binned list for a histogram of the counts 

    Parameters
    ----------
    allcounts_list : numpy.ndarray
        Output of the counts_genome function
    bar_width : float 
        It could be a function of the length of the genome e.g. bar_width=l_genome/1000

    Returns
    -------
    list
        Binned list 
    &#34;&#34;&#34;
    
    allcounts_binnedlist = []
    val_counter = 0
    sum_values = 0
    for n in range(len(allcounts_list)):
        if int(val_counter % bar_width) != 0:
            sum_values += allcounts_list[n]
        elif int(val_counter % bar_width) == 0:
            allcounts_binnedlist.append(sum_values)
            sum_values = 0
        val_counter += 1
    allcounts_binnedlist.append(sum_values)
    return allcounts_binnedlist</code></pre>
</details>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="transposonmapper.processing.profileplot_genome_helpers.binned_list"><code class="name flex">
<span>def <span class="ident">binned_list</span></span>(<span>allcounts_list, bar_width)</span>
</code></dt>
<dd>
<div class="desc"><p>A binned list for a histogram of the counts </p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>allcounts_list</code></strong> :&ensp;<code>numpy.ndarray</code></dt>
<dd>Output of the counts_genome function</dd>
<dt><strong><code>bar_width</code></strong> :&ensp;<code>float </code></dt>
<dd>It could be a function of the length of the genome e.g. bar_width=l_genome/1000</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><code>list</code></dt>
<dd>Binned list</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def binned_list(allcounts_list,bar_width):
    &#34;&#34;&#34;A binned list for a histogram of the counts 

    Parameters
    ----------
    allcounts_list : numpy.ndarray
        Output of the counts_genome function
    bar_width : float 
        It could be a function of the length of the genome e.g. bar_width=l_genome/1000

    Returns
    -------
    list
        Binned list 
    &#34;&#34;&#34;
    
    allcounts_binnedlist = []
    val_counter = 0
    sum_values = 0
    for n in range(len(allcounts_list)):
        if int(val_counter % bar_width) != 0:
            sum_values += allcounts_list[n]
        elif int(val_counter % bar_width) == 0:
            allcounts_binnedlist.append(sum_values)
            sum_values = 0
        val_counter += 1
    allcounts_binnedlist.append(sum_values)
    return allcounts_binnedlist</code></pre>
</details>
</dd>
<dt id="transposonmapper.processing.profileplot_genome_helpers.counts_genome"><code class="name flex">
<span>def <span class="ident">counts_genome</span></span>(<span>variable, bed_file, gff_file)</span>
</code></dt>
<dd>
<div class="desc"><p>Counts of reads or the transposons per chromosomes</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>variable</code></strong> :&ensp;<code>str</code></dt>
<dd>"transposons" or "reads"</dd>
<dt><strong><code>bed_file</code></strong> :&ensp;<code>str</code></dt>
<dd>absolute path of the location of the bedfile</dd>
<dt><strong><code>gff_file</code></strong> :&ensp;<code>str</code></dt>
<dd>absolute path of the location of the gff file</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><code>numpy.ndarray</code></dt>
<dd>An array of the length of the genome with the counts of each variable per location in the genome.</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def counts_genome(variable,bed_file,gff_file):
    &#34;&#34;&#34;Counts of reads or the transposons per chromosomes

    Parameters
    ----------
    variable : str
        &#34;transposons&#34; or &#34;reads&#34;
    bed_file : str
        absolute path of the location of the bedfile
    gff_file : str
        absolute path of the location of the gff file 

    Returns
    -------
    numpy.ndarray
        An array of the length of the genome with the counts of each variable per location in the genome. 
    &#34;&#34;&#34;
    
    with open(bed_file) as f:
        lines = f.readlines()
    
    chrom_names_dict, chrom_start_index_dict, chrom_end_index_dict= chromosome_name_bedfile(bed_file)
    chr_length_dict, chr_start_pos_dict, chr_end_pos_dict = chromosome_position(gff_file)
    
    summed_chr_length_dict=summed_chr(chr_length_dict)
    
    l_genome=length_genome(chr_length_dict)

    allcounts_list = np.zeros(l_genome)
    if variable == &#34;transposons&#34;:
        for line in lines[chrom_start_index_dict.get(&#34;I&#34;):chrom_end_index_dict.get(&#34;XVI&#34;)+1]:
            line = line.strip(&#39;\n&#39;).split()
            chrom_name = [k for k,v in chrom_names_dict.items() if v == line[0].replace(&#34;chr&#34;,&#39;&#39;)][0]
            allcounts_list[summed_chr_length_dict.get(chrom_name) + int(line[1])-1] += 1
    elif variable == &#34;reads&#34;:
        for line in lines[chrom_start_index_dict.get(&#34;I&#34;):chrom_end_index_dict.get(&#34;XVI&#34;)+1]:
            line = line.strip(&#39;\n&#39;).split()
            chrom_name = [k for k,v in chrom_names_dict.items() if v == line[0].replace(&#34;chr&#34;,&#39;&#39;)][0]
            allcounts_list[summed_chr_length_dict.get(chrom_name) + int(line[1])-1] += (int(line[4])-100)/20
    return allcounts_list</code></pre>
</details>
</dd>
<dt id="transposonmapper.processing.profileplot_genome_helpers.length_genome"><code class="name flex">
<span>def <span class="ident">length_genome</span></span>(<span>chr_length_dict)</span>
</code></dt>
<dd>
<div class="desc"><p>Output the length of the genome in bp </p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>chr_length_dict</code></strong> :&ensp;<code>dict</code></dt>
<dd>A dictionary describing the length of each chromosome.</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><code>int</code></dt>
<dd>The length of the genome</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def length_genome(chr_length_dict):
    
    &#34;&#34;&#34;Output the length of the genome in bp 

    Parameters
    ----------

    chr_length_dict : dict
        A dictionary describing the length of each chromosome. 

    Returns
    -------
    int
        The length of the genome
    &#34;&#34;&#34;
    
    chrom_list = [&#39;I&#39;, &#39;II&#39;, &#39;III&#39;, &#39;IV&#39;, &#39;V&#39;, &#39;VI&#39;, &#39;VII&#39;, &#39;VIII&#39;, &#39;IX&#39;, &#39;X&#39;, &#39;XI&#39;, &#39;XII&#39;, &#39;XIII&#39;, &#39;XIV&#39;, &#39;XV&#39;, &#39;XVI&#39;]

    l_genome = 0
    for chrom in chrom_list:
        l_genome += int(chr_length_dict.get(chrom))
   
    
    return l_genome</code></pre>
</details>
</dd>
<dt id="transposonmapper.processing.profileplot_genome_helpers.middle_chrom_pos"><code class="name flex">
<span>def <span class="ident">middle_chrom_pos</span></span>(<span>chr_length_dict)</span>
</code></dt>
<dd>
<div class="desc"><p>Defines the middle poit of each chromosome</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>chr_length_dict</code></strong> :&ensp;<code>dict</code></dt>
<dd>A dictionary describing the length of each chromosome.</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><code>list</code></dt>
<dd>A list describing for each chromosome the middle point.</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def middle_chrom_pos(chr_length_dict):
    &#34;&#34;&#34;Defines the middle poit of each chromosome

    Parameters
    ----------
    chr_length_dict : dict
        A dictionary describing the length of each chromosome. 

    Returns
    -------
    list
        A list describing for each chromosome the middle point. 
    &#34;&#34;&#34;
    
    summed_chr_length_dict=summed_chr(chr_length_dict)
    
    l_genome=length_genome(chr_length_dict)
    
    
    middle_chr_position = []
    c1 = summed_chr_length_dict.get(&#39;I&#39;)
    for c in summed_chr_length_dict:
        if not c == &#39;I&#39;:
            c2 = summed_chr_length_dict.get(c)
            middle_chr_position.append(c1 + (c2 - c1)/2)
            c1 = c2
            
    c2 = l_genome
    middle_chr_position.append(c1 + (c2 - c1)/2)
    
    return middle_chr_position</code></pre>
</details>
</dd>
<dt id="transposonmapper.processing.profileplot_genome_helpers.summed_chr"><code class="name flex">
<span>def <span class="ident">summed_chr</span></span>(<span>chr_length_dict)</span>
</code></dt>
<dd>
<div class="desc"><p>Create a dictionary where each value is the cumulative sum of all bp in each chromosomes</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>chr_length_dict</code></strong> :&ensp;<code>dict</code></dt>
<dd>A dictionary describing the length of each chromosome.</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><code>dict</code></dt>
<dd>A dictionary where each value corresponds to the cumulative sum of the previous chromosomes lengths.</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def summed_chr(chr_length_dict):
    &#34;&#34;&#34;Create a dictionary where each value is the cumulative sum of all bp in each chromosomes

    Parameters
    ----------
    chr_length_dict : dict
        A dictionary describing the length of each chromosome. 

    Returns
    -------
    dict
        A dictionary where each value corresponds to the cumulative sum of the previous chromosomes lengths. 
    &#34;&#34;&#34;
    
    chrom_list = [&#39;I&#39;, &#39;II&#39;, &#39;III&#39;, &#39;IV&#39;, &#39;V&#39;, &#39;VI&#39;, &#39;VII&#39;, &#39;VIII&#39;, &#39;IX&#39;, &#39;X&#39;, &#39;XI&#39;, &#39;XII&#39;, &#39;XIII&#39;, &#39;XIV&#39;, &#39;XV&#39;, &#39;XVI&#39;]
    summed_chr_length_dict = {}
    summed_chr_length = 0
    for c in chrom_list:
        summed_chr_length_dict[c] = summed_chr_length
        summed_chr_length += chr_length_dict.get(c)    
     
    return summed_chr_length_dict</code></pre>
</details>
</dd>
</dl>
</section>
<section>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="transposonmapper.processing" href="index.html">transposonmapper.processing</a></code></li>
</ul>
</li>
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="">
<li><code><a title="transposonmapper.processing.profileplot_genome_helpers.binned_list" href="#transposonmapper.processing.profileplot_genome_helpers.binned_list">binned_list</a></code></li>
<li><code><a title="transposonmapper.processing.profileplot_genome_helpers.counts_genome" href="#transposonmapper.processing.profileplot_genome_helpers.counts_genome">counts_genome</a></code></li>
<li><code><a title="transposonmapper.processing.profileplot_genome_helpers.length_genome" href="#transposonmapper.processing.profileplot_genome_helpers.length_genome">length_genome</a></code></li>
<li><code><a title="transposonmapper.processing.profileplot_genome_helpers.middle_chrom_pos" href="#transposonmapper.processing.profileplot_genome_helpers.middle_chrom_pos">middle_chrom_pos</a></code></li>
<li><code><a title="transposonmapper.processing.profileplot_genome_helpers.summed_chr" href="#transposonmapper.processing.profileplot_genome_helpers.summed_chr">summed_chr</a></code></li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html>