<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>transposonmapper.properties API documentation</title>
<meta name="description" content="" />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>transposonmapper.properties</code></h1>
</header>
<section id="section-intro">
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">from .get_coordinates_genes import get_coordinates_genes
from .get_chromosome_properties import (
    get_chromosome_names,
    get_sequence_length,
    get_chromosome_reads,
)
from .get_chromosome_position import chromosome_position
from .get_gene_position import gene_position
from .list_gene_names import (list_gene_names)
from .gene_aliases import gene_aliases

__all__ = [
    &#34;get_coordinates_genes&#34;,
    &#34;get_chromosome_names&#34;,
    &#34;get_sequence_length&#34;,
    &#34;get_chromosome_reads&#34;,
    &#34;list_gene_names&#34;,
    &#34;chromosome_position&#34;,
    &#34;gene_position&#34;,
    &#34;gene_aliases&#34;
]</code></pre>
</details>
</section>
<section>
<h2 class="section-title" id="header-submodules">Sub-modules</h2>
<dl>
<dt><code class="name"><a title="transposonmapper.properties.get_chromosome_position" href="get_chromosome_position.html">transposonmapper.properties.get_chromosome_position</a></code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt><code class="name"><a title="transposonmapper.properties.get_chromosome_properties" href="get_chromosome_properties.html">transposonmapper.properties.get_chromosome_properties</a></code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt><code class="name"><a title="transposonmapper.properties.get_gene_position" href="get_gene_position.html">transposonmapper.properties.get_gene_position</a></code></dt>
<dd>
<div class="desc"></div>
</dd>
</dl>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="transposonmapper.properties.chromosome_position"><code class="name flex">
<span>def <span class="ident">chromosome_position</span></span>(<span>gff_file)</span>
</code></dt>
<dd>
<div class="desc"><p>Get the start and end position of each chromosome and determine their respective length.
Input is a .gff file downloaded from <a href="https://www.ensembl.org/Saccharomyces_cerevisiae/Info/Index">https://www.ensembl.org/Saccharomyces_cerevisiae/Info/Index</a>
Output are three dictionaries for length, start and end position. All dictionaries have keys representing the chromosome number in roman numerals.
To get all dictionaries, use: 'a,b,c, chromosome_and_gene_position.chromosome_position()'.
'a' = chromosome length
'b' = chromosome start position
'c' = chromosome end position</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def chromosome_position(gff_file):
    &#34;&#34;&#34;Get the start and end position of each chromosome and determine their respective length.
    Input is a .gff file downloaded from https://www.ensembl.org/Saccharomyces_cerevisiae/Info/Index
    Output are three dictionaries for length, start and end position. All dictionaries have keys representing the chromosome number in roman numerals.
    To get all dictionaries, use: &#39;a,b,c, chromosome_and_gene_position.chromosome_position()&#39;.
    &#39;a&#39; = chromosome length
    &#39;b&#39; = chromosome start position
    &#39;c&#39; = chromosome end position
    &#34;&#34;&#34;

    if gff_file==None:
        default_path = pkg_resources.resource_filename(&#34;transposonmapper&#34;, &#34;data_files/&#34;)
        gff_file = os.path.join(
            default_path, &#34;Saccharomyces_cerevisiae.R64-1-1.99.gff3&#34;
        )
    # Get roman to arabic dictionary
    roman_to_arabic = chromosomename_roman_to_arabic()[1]

    # GET END POSITIONS OF THE CHROMOSOMES FROM THE GFF FILE AND STORE THEM IN A DICTIONARY
    chr_length_dict = {}
    chr_length_list = []
    with open(gff_file) as f:
        line_counter = 0
        next(f)
        while line_counter &lt; 17:
            lines = f.readline()
            chr_line_list = lines.strip(&#34;\n&#34;).replace(&#34; &#34;, &#34;\t&#34;).split(&#34;\t&#34;)
            chr_number = chr_line_list[3]
            # if chr_number != &#39;Mito&#39;:
            chr_length = int(chr_line_list[5])
            chr_length_list.append(chr_length)
            chr_length_dict[chr_number] = chr_length
            line_counter += 1

    # DETERMINE START AND END POSITION OF EACH OF THE CHROMOSOMES
    chr_start_pos_dict = {}
    chr_end_pos_dict = {}
    counter = 0
    for roman in roman_to_arabic.keys():
        chr_start_pos_dict[roman] = sum(chr_length_list[:counter]) + 1
        chr_end_pos_dict[roman] = sum(chr_length_list[: counter + 1])
        if roman == &#34;I&#34;:
            chr_start_pos_dict[roman] = 1
        counter += 1

    return (chr_length_dict, chr_start_pos_dict, chr_end_pos_dict)</code></pre>
</details>
</dd>
<dt id="transposonmapper.properties.gene_aliases"><code class="name flex">
<span>def <span class="ident">gene_aliases</span></span>(<span>gene_information_file=None)</span>
</code></dt>
<dd>
<div class="desc"><p>Create three dictionaries containing aliases for genes
Input is the path to 'Protein_Names.txt' file downloaded from <a href="https://www.uniprot.org/docs/yeast.">https://www.uniprot.org/docs/yeast.</a>
If no input is given the file is automatically searched for at thisscriptlocation/../Data_Files/Yeast_Protein_Names.txt.
Output is three dictionaries:
aliases_designation_dict = gene aliases for common names (e.g. Bem1 and Sro1)
aliases_sgd_dict = gene aliases for the search names in SGD (e.g. Bem1 and S000000404)
aliases_swissprot_dict = gene aliases for Swiss Prot.
The keys of the dictionaries are the systematic names of the genes (e.g. YBR200W for Bem1)</p>
<p>Search through lists to get corresponding key:
[key for key, val in aliases.items() if 'TFC3' in val]</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def gene_aliases(gene_information_file=None):
    &#34;&#34;&#34;Create three dictionaries containing aliases for genes
    Input is the path to &#39;Protein_Names.txt&#39; file downloaded from https://www.uniprot.org/docs/yeast.
    If no input is given the file is automatically searched for at thisscriptlocation/../Data_Files/Yeast_Protein_Names.txt.
    Output is three dictionaries:
    aliases_designation_dict = gene aliases for common names (e.g. Bem1 and Sro1)
    aliases_sgd_dict = gene aliases for the search names in SGD (e.g. Bem1 and S000000404)
    aliases_swissprot_dict = gene aliases for Swiss Prot.
    The keys of the dictionaries are the systematic names of the genes (e.g. YBR200W for Bem1)
    
    Search through lists to get corresponding key:
        [key for key, val in aliases.items() if &#39;TFC3&#39; in val]
    &#34;&#34;&#34;

    if gene_information_file == None:

        default_path = pkg_resources.resource_filename(&#34;transposonmapper&#34;, &#34;data_files/&#34;)
    
        gene_information_file = os.path.join(default_path, &#34;Yeast_Protein_Names.txt&#34;)
        

        

    aliases_designation_dict = {}
    aliases_sgd_dict = {}
    aliases_swissprot_dict = {}

    with open(gene_information_file) as f:
        lines = f.readlines()

        for i in range(
            58, len(lines) - 6
        ):  # THE GENES START AT LINE 58 AND STOP 6 LINES BEFORE THE END OF THE FILE.
            l = lines[i]
            l_list = l.split()

            alias_counter = 0
            gene_designation_list = []
            for names in l_list:
                if names.endswith(&#34;;&#34;):
                    gene_designation_list.append(names.strip(&#34;;&#34;))
                    alias_counter += 1
            gene_designation_list.append(l_list[alias_counter])

            oln_name = l_list[alias_counter + 1]
            sgd_name = l_list[alias_counter + 4]
            swissprot_name = l_list[alias_counter + 2]
            if oln_name == &#34;GAG&#34; or oln_name == &#34;POL&#34;:
                oln_name = l_list[alias_counter + 2]
                sgd_name = l_list[alias_counter + 5]
                swissprot_name = l_list[alias_counter + 3]

            aliases_designation_dict[oln_name] = gene_designation_list
            aliases_sgd_dict[oln_name] = sgd_name
            aliases_swissprot_dict[oln_name] = swissprot_name

    return (aliases_designation_dict, aliases_sgd_dict, aliases_swissprot_dict)</code></pre>
</details>
</dd>
<dt id="transposonmapper.properties.gene_position"><code class="name flex">
<span>def <span class="ident">gene_position</span></span>(<span>gff_file, get_dict=True)</span>
</code></dt>
<dd>
<div class="desc"><p>Get the start and end position of each gene and determine their respective length.
Input is a .gff file downloaded from <a href="https://www.ensembl.org/Saccharomyces_cerevisiae/Info/Index">https://www.ensembl.org/Saccharomyces_cerevisiae/Info/Index</a>
Output is a dictionary that includes all gene names as keys. The values are lists with four inputs.
The first is the chromosome number the gene belong to, the second is the start position, the third is the end position of the gene in terms of basepairs, the fourth is the reading orientation of the gene.
The reading orientation is indicated with a '+' (forward reading) or '-' (reverse reading).
The get_dict by defulat sets that the output should be given as a dictionary with keys the different genes and the values a list of the different parameters.
When the get_dict is set to False, the code returns all the values as individual lists.</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def gene_position(gff_file, get_dict=True):
    &#34;&#34;&#34;Get the start and end position of each gene and determine their respective length.
    Input is a .gff file downloaded from https://www.ensembl.org/Saccharomyces_cerevisiae/Info/Index
    Output is a dictionary that includes all gene names as keys. The values are lists with four inputs.
    The first is the chromosome number the gene belong to, the second is the start position, the third is the end position of the gene in terms of basepairs, the fourth is the reading orientation of the gene.
    The reading orientation is indicated with a &#39;+&#39; (forward reading) or &#39;-&#39; (reverse reading).
    The get_dict by defulat sets that the output should be given as a dictionary with keys the different genes and the values a list of the different parameters.
    When the get_dict is set to False, the code returns all the values as individual lists.
    &#34;&#34;&#34;

    if get_dict == True:
        gene_pos_dict = {}
        with open(gff_file) as f:
            for line in f:
                line_list = line.split(&#34;\t&#34;)
                if len(line_list) &gt; 2:
                    if line_list[2] == &#34;gene&#34;:
                        gene_chr = line_list[0]
                        gene_start = line_list[3]
                        gene_end = line_list[4]
                        gene_orien = line_list[6]
                        gene_position = [
                            gene_chr,
                            int(gene_start),
                            int(gene_end),
                            gene_orien,
                        ]

                        gene_name_string = line_list[8].split(&#34;;&#34;)[0]
                        gene_name = gene_name_string.split(&#34;:&#34;)[1]

                        gene_pos_dict[gene_name] = gene_position

        return gene_pos_dict

    else:
        gene_chr = []
        gene_start = []
        gene_end = []
        gene_orien = []
        gene_name = []
        with open(gff_file) as f:
            for line in f:
                line_list = line.split(&#34;\t&#34;)
                if len(line_list) &gt; 2:
                    if line_list[2] == &#34;gene&#34;:
                        gene_chr.append(line_list[0])
                        gene_start.append(int(line_list[3]))
                        gene_end.append(int(line_list[4]))
                        gene_orien.append(line_list[6])

                        gene_name_string = line_list[8].split(&#34;;&#34;)[0]
                        gene_name.append(gene_name_string.split(&#34;:&#34;)[1])

        return (gene_name, gene_chr, gene_start, gene_end, gene_orien)</code></pre>
</details>
</dd>
<dt id="transposonmapper.properties.get_chromosome_names"><code class="name flex">
<span>def <span class="ident">get_chromosome_names</span></span>(<span>bam)</span>
</code></dt>
<dd>
<div class="desc"><p>This functions translate the format of the chromosome names
from the alignment file from the pysam module into numbers</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>bam</code></strong> :&ensp;<code>dict , It is the output</code> of <code>the function
pysam.AlignmentFile(bamfile, "rb")</code></dt>
<dd>&nbsp;</dd>
</dl>
<h2 id="returns">Returns</h2>
<p>ref_tid : dict, where the values are the chromosome numbers per key in the bam file.</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def get_chromosome_names(bam):
    &#34;&#34;&#34; This functions translate the format of the chromosome names
    from the alignment file from the pysam module into numbers
    

    Parameters
    ----------
    bam : dict , It is the output of the function  pysam.AlignmentFile(bamfile, &#34;rb&#34;)    

    Returns
    -------
    ref_tid : dict, where the values are the chromosome numbers per key in the bam file. 
    
    &#34;&#34;&#34;

    # ref_tid = {str(name): int(bam.get_tid(name)) + 1 for name in bam.get_reference_name}

    ref_tid = {} # &#39;I&#39; | 0, &#39;II&#39; | 1, ...
    for i in range(bam.nreferences): #if bam.nreferences does not work, use range(17) #16 chromosomes and the mitochondrial chromosome
        ref_name = bam.get_reference_name(i)
        ref_tid[ref_name] = bam.get_tid(ref_name)

    return ref_tid</code></pre>
</details>
</dd>
<dt id="transposonmapper.properties.get_chromosome_reads"><code class="name flex">
<span>def <span class="ident">get_chromosome_reads</span></span>(<span>bam)</span>
</code></dt>
<dd>
<div class="desc"><p>This function returns statistics about mapped/unmapped reads per chromosome as they are stored in the index.
It makes use of the method get_index_statistics() from the pysam module.
Parameters</p>
<hr>
<dl>
<dt><strong><code>bam</code></strong> :&ensp;<code>dict, It is the output</code> of <code>the function
pysam.AlignmentFile(bamfile, "rb")</code></dt>
<dd>&nbsp;</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>mapped_reads</code></strong> :&ensp;<code>dict[str, list]</code></dt>
<dd>Syntax 'I' | [mapped, unmapped, total reads]</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def get_chromosome_reads(bam):
    &#34;&#34;&#34; This function returns statistics about mapped/unmapped reads per chromosome as they are stored in the index.
    It makes use of the method get_index_statistics() from the pysam module. 
    Parameters
    ----------
    bam : dict, It is the output of the function  pysam.AlignmentFile(bamfile, &#34;rb&#34;)              

    Returns
    -------
    mapped_reads : dict[str, list]
        Syntax &#39;I&#39; | [mapped, unmapped, total reads]

    &#34;&#34;&#34;
    stats = bam.get_index_statistics()
    mapped_reads = {}
    for stat in stats:
        mapped_reads[stat[0]] = [stat[1], stat[2], stat[3]]
        if stat[2] != 0:
            warnings.warn(&#34;Unmapped reads found in chromosome &#34; + stat[0])

    return mapped_reads</code></pre>
</details>
</dd>
<dt id="transposonmapper.properties.get_coordinates_genes"><code class="name flex">
<span>def <span class="ident">get_coordinates_genes</span></span>(<span>path: str = '', data_files: dict = {})</span>
</code></dt>
<dd>
<div class="desc"><p>Get coordinates of all genes</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>path</code></strong> :&ensp;<code>str</code></dt>
<dd>&nbsp;</dd>
<dt><strong><code>data_files</code></strong> :&ensp;<code>dict</code></dt>
<dd>&nbsp;</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>essential_coordinates</code></strong> :&ensp;<code>dict
</code></dt>
<dd>&nbsp;</dd>
<dt><strong><code>aliases_designation</code></strong> :&ensp;<code>dict</code></dt>
<dd>&nbsp;</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def get_coordinates_genes(path: str = &#34;&#34;, data_files: dict = {}):
    &#34;&#34;&#34;
    Get coordinates of all genes

    Parameters
    ----------
    path : str
    data_files : dict

    Returns
    -------
    essential_coordinates : dict    
    aliases_designation : dict

    &#34;&#34;&#34;

    essential_coordinates = {}

    # Get position genes
    if &#34;gff3&#34; in data_files:
        file_path = os.path.join(path, data_files[&#34;gff3&#34;])
        gene_coordinates = gene_position(file_path)
    else:
        raise ValueError(&#34;gff3 type not found in data&#34;)

    # Get all annotated essential genes
    if &#34;essential_genes&#34; in data_files:
        file_path = os.path.join(path, data_files[&#34;essentials&#34;])
        with open(file_path, &#34;r&#34;) as f:
            genes = f.readlines()[1:]
            for gene in genes:
                name = gene.strip(&#34;\n&#34;)
                essential_coordinates[name] = gene_coordinates.get(name).copy()
    else:
        raise ValueError(&#34;essentials not found in data&#34;)

    # Get aliases of all genes
    if &#34;gene_names&#34; in data_files:
        file_path = os.path.join(path, &#34;Yeast_Protein_Names.txt&#34;)
        aliases_designation = gene_aliases(file_path)[0]  #&#39;YMR056C&#39; \ [&#39;AAC1&#39;], ...
    else:
        raise ValueError(&#34;gene_names not found in data&#34;)

    return essential_coordinates, aliases_designation</code></pre>
</details>
</dd>
<dt id="transposonmapper.properties.get_sequence_length"><code class="name flex">
<span>def <span class="ident">get_sequence_length</span></span>(<span>bam)</span>
</code></dt>
<dd>
<div class="desc"><p>This function returns the length of each chromosome and a cumulative sum of them.</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>bam</code></strong> :&ensp;<code>dict, It is the output</code> of <code>the function
pysam.AlignmentFile(bamfile, "rb")</code></dt>
<dd>&nbsp;</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>chr_lengths</code></strong> :&ensp;<code>dict, A dictionary which very value is the length in basepairs</code></dt>
<dd>&nbsp;</dd>
<dt><code>of each chromosome in the alignment file. </code></dt>
<dd>&nbsp;</dd>
<dt><strong><code>chr_lengths_cumsum</code></strong> :&ensp;<code>dict , A dictionary which very value is the sum</code> of <code>the length in basepairs</code></dt>
<dd>&nbsp;</dd>
<dt><code>of the previous chromosomes in the alignment file. chr_lengths_cumsum{n}=cumulative sum (n-1 previous chromosomes</code></dt>
<dd>&nbsp;</dd>
<dt><code>length)</code></dt>
<dd>&nbsp;</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def get_sequence_length(bam):
    &#34;&#34;&#34; This function returns the length of each chromosome and a cumulative sum of them.
    
    Parameters
    ----------
    bam : dict, It is the output of the function  pysam.AlignmentFile(bamfile, &#34;rb&#34;)          

    Returns
    -------
    chr_lengths : dict, A dictionary which very value is the length in basepairs
    of each chromosome in the alignment file. 
    chr_lengths_cumsum : dict , A dictionary which very value is the sum of the length in basepairs
    of the previous chromosomes in the alignment file. chr_lengths_cumsum{n}=cumulative sum (n-1 previous chromosomes
    length)

    &#34;&#34;&#34;

    chr_lengths = {}  # &#39;I&#39; | 230218, &#39;II&#39; | 813184, ...
    chr_lengths_cumsum = {}  # &#39;I&#39; | 0, &#39;II&#39; | 230218, &#39;III&#39; |  1043402, ...
    ref_summedlength = 0
    ref_tid = get_chromosome_names(bam)
    for key in ref_tid.keys():
        ref_length = bam.get_reference_length(key)
        chr_lengths[key] = bam.get_reference_length(key)
        chr_lengths_cumsum[key] = ref_summedlength
        ref_summedlength += ref_length

    return chr_lengths, chr_lengths_cumsum</code></pre>
</details>
</dd>
<dt id="transposonmapper.properties.list_gene_names"><code class="name flex">
<span>def <span class="ident">list_gene_names</span></span>(<span>gene_information_file=None)</span>
</code></dt>
<dd>
<div class="desc"><p>Create a list of all known gene names and their aliases as listed on SGD (or as provided as an optional input file)
Input is a standard file downloaded from <a href="https://www.uniprot.org/docs/yeast.">https://www.uniprot.org/docs/yeast.</a>
Output is list of all genes, which also includes all the aliases (if they exists).</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def list_gene_names(gene_information_file=None):
    &#34;&#34;&#34;Create a list of all known gene names and their aliases as listed on SGD (or as provided as an optional input file)
    Input is a standard file downloaded from https://www.uniprot.org/docs/yeast.
    Output is list of all genes, which also includes all the aliases (if they exists).
    &#34;&#34;&#34;

    if gene_information_file == None:
       
        default_path = pkg_resources.resource_filename(&#34;transposonmapper&#34;, &#34;data_files/&#34;)
    
        gene_information_file = os.path.join(
            default_path, &#34;Yeast_Protein_Names.txt&#34;
        )
        
    gene_name_list = []  # INLCUDES ALL GENE NAMES AND POTENTIAL ALIASES
    gene_oln_list = []  # INCLUDE ONLY THE OLN NAMING CONVENTION
    gene_sgd_list = []  # INCLUDE THE FIRST DESIGNATION NAME THAT IS IN THE INPUT LIST
    gene_counter = 0
    with open(gene_information_file) as f:
        lines = f.readlines()
        for i in range(
            58, len(lines) - 6
        ):  # THE GENES START AT LINE 58 AND STOP 6 LINES BEFORE THE END OF THE FILE.
            n = 0
            l = lines[i]

            extra_columns = l.count(
                &#34;;&#34;
            )  # COUNT HOW MANY TIMES &#39;;&#39; OCCURS IN A LINE. THIS IS NEEDED TO GET THE RIGHT COLUMNS AS SOMETIMES ALIASES OF GENES ARE PRESENTED IN EXTRA COLUMNS
            l_short = &#34; &#34;.join(l.split())
            l_list = l_short.split(&#34; &#34;)

            gene_name_list.append(l_list[0].strip(&#34;;&#34;))
            gene_sgd_list.append(l_list[0].strip(&#34;;&#34;))

            gene_oln = l_list[1 + extra_columns].strip(&#34;;&#34;)  # GET THE OLN NAME
            if (
                gene_oln == &#34;GAG&#34; or gene_oln == &#34;POL&#34;
            ):  # CHECK WHETHER THE OLN IS &#39;GAG&#39; OR &#39;POL&#39;. IF YES, TAKE THE NEXT COLUMN
                gene_name_list.append(l_list[2 + extra_columns].strip(&#34;;&#34;))
                gene_oln_list.append(l_list[2 + extra_columns].strip(&#34;;&#34;))
            else:
                gene_name_list.append(gene_oln)
                gene_oln_list.append(gene_oln)

            if (
                l_list[1 + extra_columns] == &#34;GAG&#34; or l_list[1 + extra_columns] == &#34;POL&#34;
            ):  # THESE ARE SEQUENCES THAT SOMETIMES OCCUR WHICH HAVE TO BE IGNORED.
                extra_columns = extra_columns + 1
            if extra_columns &gt; 0:
                for n in range(extra_columns):
                    gene_name = l_list[1 + n].strip(&#34;;&#34;)
                    if not gene_name == &#34;GAG&#34; and not gene_name == &#34;POL&#34;:
                        gene_name_list.append(gene_name)
            gene_counter += 1

    ###SAVING OLN LIST
    #    from datetime import date
    #    current_date = date.today()

    #    oln_saving_file = r&#34;S_Cerevisiae_protein_oln_name_full_genome.txt&#34;
    #    gene_oln_list_sorted = sorted(gene_oln_list)
    #    with open(oln_saving_file, &#39;w&#39;) as f:
    #
    ##        f.write(&#34;org=S. Cerevisiae ; type=Genomic ; naming=&#39;oln&#39; ; source file=&#39;Yeast_Protein_Names.txt&#39; ; creation date=%s using &#39;gene_names.py&#39;\n&#34; % current_date)
    #        for oln_name in gene_oln_list_sorted:
    #            f.write(&#34;%s\n&#34; % oln_name)

    #    sgd_saving_file = r&#34;S_Cerevisiae_protein_designation_name_full_genome.txt&#34;
    #    gene_sgd_list_sorted = sorted(gene_sgd_list)
    #    with open(sgd_saving_file, &#39;w&#39;) as f:
    #
    #        f.write(&#34;org=S. Cerevisiae ; type=Genomic ; naming=&#39;designation&#39; ; source file=&#39;Yeast_Protein_Names.txt&#39; ; creation date=%s using &#39;gene_names.py&#39;\n&#34; % current_date)
    #        for sgd_name in gene_sgd_list_sorted:
    #            f.write(&#34;%s \n&#34; % sgd_name)

    ###
    print(&#34;Number of genes found in file = &#34;, gene_counter)
    return gene_name_list</code></pre>
</details>
</dd>
</dl>
</section>
<section>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="transposonmapper" href="../index.html">transposonmapper</a></code></li>
</ul>
</li>
<li><h3><a href="#header-submodules">Sub-modules</a></h3>
<ul>
<li><code><a title="transposonmapper.properties.get_chromosome_position" href="get_chromosome_position.html">transposonmapper.properties.get_chromosome_position</a></code></li>
<li><code><a title="transposonmapper.properties.get_chromosome_properties" href="get_chromosome_properties.html">transposonmapper.properties.get_chromosome_properties</a></code></li>
<li><code><a title="transposonmapper.properties.get_gene_position" href="get_gene_position.html">transposonmapper.properties.get_gene_position</a></code></li>
</ul>
</li>
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="">
<li><code><a title="transposonmapper.properties.chromosome_position" href="#transposonmapper.properties.chromosome_position">chromosome_position</a></code></li>
<li><code><a title="transposonmapper.properties.gene_aliases" href="#transposonmapper.properties.gene_aliases">gene_aliases</a></code></li>
<li><code><a title="transposonmapper.properties.gene_position" href="#transposonmapper.properties.gene_position">gene_position</a></code></li>
<li><code><a title="transposonmapper.properties.get_chromosome_names" href="#transposonmapper.properties.get_chromosome_names">get_chromosome_names</a></code></li>
<li><code><a title="transposonmapper.properties.get_chromosome_reads" href="#transposonmapper.properties.get_chromosome_reads">get_chromosome_reads</a></code></li>
<li><code><a title="transposonmapper.properties.get_coordinates_genes" href="#transposonmapper.properties.get_coordinates_genes">get_coordinates_genes</a></code></li>
<li><code><a title="transposonmapper.properties.get_sequence_length" href="#transposonmapper.properties.get_sequence_length">get_sequence_length</a></code></li>
<li><code><a title="transposonmapper.properties.list_gene_names" href="#transposonmapper.properties.list_gene_names">list_gene_names</a></code></li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html>