<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>transposonmapper.exporting API documentation</title>
<meta name="description" content="" />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>transposonmapper.exporting</code></h1>
</header>
<section id="section-intro">
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">from .save_as_bed import save_as_bed
from .save_per_gene import save_per_gene
from .save_per_gene_insertions import (
    save_per_essential_insertions,
    save_per_gene_insertions,
)
from .save_as_wig import save_as_wig

__all__ = [
    &#34;save_as_wig&#34;,
    &#34;save_as_bed&#34;,
    &#34;save_per_gene&#34;,
    &#34;save_per_essential_insertions&#34;,
    &#34;save_per_gene_insertions&#34;,
]</code></pre>
</details>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="transposonmapper.exporting.save_as_bed"><code class="name flex">
<span>def <span class="ident">save_as_bed</span></span>(<span>file, tncoordinates_array, ref_tid, readnumb_array)</span>
</code></dt>
<dd>
<div class="desc"><p>This function writes in a .bed file located in the same folder as the bamfile, the information abot how many transposons
were found in each genomic location. </p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>file</code></strong> :&ensp;<code>str</code></dt>
<dd>Path with the filename extension included(e.g "data_file/bedfile_name.bed") describing where do you want to store the results.
By default it will be stored in the same location as the bamfile, with the same basename.
Example, if the bamfile path is data_file/data_1.bam then the bed file will be data_file/data_1.bam.bed</dd>
<dt><strong><code>tncoordinates_array</code></strong> :&ensp;<code>numpy array </code></dt>
<dd>Second Output from the get_reads.py function : <em>,tncoordinates_array,</em>=get_reads(bam)</dd>
<dt><strong><code>ref_tid</code></strong> :&ensp;<code>dict </code></dt>
<dd>Output from the get_chromosome_names function : ref_tid = get_chromosome_names(bam)</dd>
<dt><strong><code>readnumb_array</code></strong> :&ensp;<code>numpy array </code></dt>
<dd>1st Output from the get_reads.py function : readnumb_array,<em>,</em>=get_reads(bam)</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def save_as_bed(file, tncoordinates_array, ref_tid, readnumb_array):
    &#34;&#34;&#34;This function writes in a .bed file located in the same folder as the bamfile, the information abot how many transposons
    were found in each genomic location. 

    Parameters
    ----------
    file : str
        Path with the filename extension included(e.g &#34;data_file/bedfile_name.bed&#34;) describing where do you want to store the results.
        By default it will be stored in the same location as the bamfile, with the same basename. 
        Example, if the bamfile path is data_file/data_1.bam then the bed file will be data_file/data_1.bam.bed 
    tncoordinates_array : numpy array 
        Second Output from the get_reads.py function : _,tncoordinates_array,_=get_reads(bam)
    ref_tid : dict 
        Output from the get_chromosome_names function : ref_tid = get_chromosome_names(bam)
    readnumb_array : numpy array 
        1st Output from the get_reads.py function : readnumb_array,_,_=get_reads(bam)
    &#34;&#34;&#34;

    with open(file, &#34;w&#34;) as f:

        base = os.path.basename(file)
        filename = os.path.splitext(base)[0]
        f.write(&#34;track name=&#34; + filename + &#34; useScore=1\n&#34;)

        coordinates_counter = 0
        for tn in tncoordinates_array:
            refname = [key for key, val in ref_tid.items() if val == tn[0] - 1][0]
            if refname == &#34;Mito&#34;:
                refname = &#34;M&#34;
            f.write(
                &#34;chr&#34;
                + refname
                + &#34; &#34;
                + str(tn[1])
                + &#34; &#34;
                + str(tn[1] + 1)
                + &#34; . &#34;
                + str(100 + readnumb_array[coordinates_counter] * 20)
                + &#34;\n&#34;
            )
            coordinates_counter += 1</code></pre>
</details>
</dd>
<dt id="transposonmapper.exporting.save_as_wig"><code class="name flex">
<span>def <span class="ident">save_as_wig</span></span>(<span>wigfile, tncoordinates_array, ref_tid, readnumb_array)</span>
</code></dt>
<dd>
<div class="desc"><p>This function writes in a .wig file located in the same folder as the bamfile, the information abot how many transposons
were found in each genomic location. </p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>wigfile</code></strong> :&ensp;<code>str </code></dt>
<dd>Path with the filename extension included(e.g "data_file/wigfile_name.wig") describing where do you want to store the results.
By default it will be stored in the same location as the bamfile, with the same basename.
Example, if the bamfile path is data_file/data_1.bam then the wig file will be data_file/data_1.bam.wig</dd>
<dt><strong><code>tncoordinates_array</code></strong> :&ensp;<code>numpy array </code></dt>
<dd>Second Output from the get_reads.py function : <em>,tncoordinates_array,</em>=get_reads(bam)</dd>
<dt><strong><code>ref_tid</code></strong> :&ensp;<code>dict </code></dt>
<dd>Output from the get_chromosome_names function : ref_tid = get_chromosome_names(bam)</dd>
<dt><strong><code>readnumb_array</code></strong> :&ensp;<code>numpy array </code></dt>
<dd>1st Output from the get_reads.py function : readnumb_array,<em>,</em>=get_reads(bam)</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def save_as_wig(wigfile, tncoordinates_array, ref_tid, readnumb_array):
    &#34;&#34;&#34;This function writes in a .wig file located in the same folder as the bamfile, the information abot how many transposons
    were found in each genomic location. 

    Parameters
    ----------
    wigfile : str 
        Path with the filename extension included(e.g &#34;data_file/wigfile_name.wig&#34;) describing where do you want to store the results.
        By default it will be stored in the same location as the bamfile, with the same basename. 
        Example, if the bamfile path is data_file/data_1.bam then the wig file will be data_file/data_1.bam.wig
    tncoordinates_array : numpy array 
        Second Output from the get_reads.py function : _,tncoordinates_array,_=get_reads(bam)
    ref_tid : dict 
        Output from the get_chromosome_names function : ref_tid = get_chromosome_names(bam)
    readnumb_array : numpy array 
        1st Output from the get_reads.py function : readnumb_array,_,_=get_reads(bam)
    &#34;&#34;&#34;

    readnumbwig_array = readnumb_array
    ref_names = list(ref_tid.keys())

    unique_index_array = np.array([], dtype=int)  # =cc
    N_uniques_perchr_list = []
    ll = 0
    for kk in ref_names:
        # get indices for current chromosome.
        index = np.where(tncoordinates_array[:, 0] == int(ref_tid[kk] + 1))

        # get all insertion locations (in tncoordinates, all rows, column 1)
        unique_index = np.unique(tncoordinates_array[index][:, 1], return_index=True)[1]

        unique_index_array = np.append(unique_index_array, (unique_index + ll), axis=0)

        # total amount unique indices found untill current chromosome
        ll += np.count_nonzero(tncoordinates_array[:, 0] == int(ref_tid[kk] + 1))
        N_uniques_perchr_list.append(ll)

    # Collect duplicates
    duplicate_list = []  # =dd
    ll = 0
    index_last_unique_previous_chromosome = 0
    for ii in N_uniques_perchr_list:
        index_last_unique = np.where(unique_index_array &lt;= ii)[0][-1]
        for jj in range(ll, ii):
            if (
                int(jj)
                not in unique_index_array[
                    index_last_unique_previous_chromosome:index_last_unique
                ]
            ):
                duplicate_list.append(jj)
        index_last_unique_previous_chromosome = index_last_unique
        ll = ii

    # SUM READNUMB VALUES AT INDEX IN DUPLICATE_LIST AND DUPLICATE_LIST-1
    for ii in duplicate_list:
        readnumbwig_array[ii - 1] = readnumbwig_array[ii - 1] + readnumbwig_array[ii]

    tncoordinateswig_duplicatesremoved_array = np.delete(
        tncoordinates_array, duplicate_list, axis=0
    )
    readnumbwig_duplicatesremoved_array = np.delete(
        readnumbwig_array, duplicate_list, axis=0
    )

    # Write wigfile
    with open(wigfile, &#34;w&#34;) as f:
        base = os.path.basename(wigfile)
        filename = os.path.splitext(base)[0]
        f.write(&#34;track type=wiggle_0 ,maxheightPixels=60 name=&#34; + filename + &#34;\n&#34;)
        for kk in ref_names:
            f.write(&#34;VariableStep chrom=chr&#34; + kk + &#34;\n&#34;)

            index = np.where(
                tncoordinateswig_duplicatesremoved_array[:, 0] == int(ref_tid[kk] + 1)
            )  # get indices for current chromosome.
            for ii in index[0]:
                f.write(
                    str(tncoordinateswig_duplicatesremoved_array[ii][1])
                    + &#34; &#34;
                    + str(readnumbwig_duplicatesremoved_array[ii])
                    + &#34;\n&#34;
                )</code></pre>
</details>
</dd>
<dt id="transposonmapper.exporting.save_per_essential_insertions"><code class="name flex">
<span>def <span class="ident">save_per_essential_insertions</span></span>(<span>filename, tn_coordinates, gene_coordinates, chr_lengths_cumsum, ref_tid_roman, aliases_designation)</span>
</code></dt>
<dd>
<div class="desc"></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def save_per_essential_insertions(
    filename,
    tn_coordinates,
    gene_coordinates,
    chr_lengths_cumsum,
    ref_tid_roman,
    aliases_designation,
):
    &#34;&#34;&#34;

    
    &#34;&#34;&#34;

    with open(filename, &#34;w&#34;) as f:

        f.write(
            &#34;Essential gene name\tChromosome\tStart location\tEnd location\tInsertion locations\tReads per insertion location\n&#34;
        )

        for gene in tn_coordinates:
            gene_chrom = ref_tid_roman.get(gene_coordinates.get(gene)[0])
            tncoordinates = [
                ins - chr_lengths_cumsum.get(gene_chrom)
                for ins in tn_coordinates[gene][3]
            ]

            if gene in aliases_designation:
                gene_alias = aliases_designation.get(gene)[0]
            else:
                gene_alias = gene

            f.write(
                gene_alias
                + &#34;\t&#34;
                + str(tn_coordinates[gene][0])
                + &#34;\t&#34;
                + str(tn_coordinates[gene][1] - chr_lengths_cumsum.get(gene_chrom))
                + &#34;\t&#34;
                + str(tn_coordinates[gene][2] - chr_lengths_cumsum.get(gene_chrom))
                + &#34;\t&#34;
                + str(tncoordinates)
                + &#34;\t&#34;
                + str(tn_coordinates[gene][4])
                + &#34;\n&#34;
            )</code></pre>
</details>
</dd>
<dt id="transposonmapper.exporting.save_per_gene"><code class="name flex">
<span>def <span class="ident">save_per_gene</span></span>(<span>filename, tn_per_gene, reads_per_gene, aliases_designation)</span>
</code></dt>
<dd>
<div class="desc"><p>Create text file with transposons and reads per gene</p>
<p>NOTE THAT THE TRANSPOSON WITH THE HIGHEST READ COUNT IS IGNORED.
E.G. IF THIS FILE IS COMPARED WITH THE _PERGENE_INSERTIONS.TXT FILE THE
READS DON'T ADD UP (SEE <a href="https://groups.google.com/forum/#!category-topic/satayusers/bioinformatics/uaTpKsmgU6Q">https://groups.google.com/forum/#!category-topic/satayusers/bioinformatics/uaTpKsmgU6Q</a>)
TOO REMOVE THIS HACK, CHANGE THE INITIALIZATION OF THE VARIABLE readpergene</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def save_per_gene(filename, tn_per_gene, reads_per_gene, aliases_designation):
    &#34;&#34;&#34;
    Create text file with transposons and reads per gene
 
    NOTE THAT THE TRANSPOSON WITH THE HIGHEST READ COUNT IS IGNORED.
    E.G. IF THIS FILE IS COMPARED WITH THE _PERGENE_INSERTIONS.TXT FILE THE 
    READS DON&#39;T ADD UP (SEE https://groups.google.com/forum/#!category-topic/satayusers/bioinformatics/uaTpKsmgU6Q)
    TOO REMOVE THIS HACK, CHANGE THE INITIALIZATION OF THE VARIABLE readpergene
 
    &#34;&#34;&#34;

    with open(filename, &#34;w&#34;) as f:

        f.write(&#34;Gene name\tNumber of transposons per gene\tNumber of reads per gene\n&#34;)

        for gene in tn_per_gene:
            tnpergene = tn_per_gene[gene]
            readpergene = reads_per_gene[gene]
            if gene in aliases_designation:
                gene_alias = aliases_designation.get(gene)[0]
            else:
                gene_alias = gene
            f.write(gene_alias + &#34;\t&#34; + str(tnpergene) + &#34;\t&#34; + str(readpergene) + &#34;\n&#34;)</code></pre>
</details>
</dd>
<dt id="transposonmapper.exporting.save_per_gene_insertions"><code class="name flex">
<span>def <span class="ident">save_per_gene_insertions</span></span>(<span>filename, tn_coordinates, gene_coordinates, chr_lengths_cumsum, ref_tid_roman, aliases_designation)</span>
</code></dt>
<dd>
<div class="desc"><p>This function write in txt file 5 columns of information about transposon insertions separated by tabs.
The tabs are: Gene name,Chromosome,Start location,End location,Insertion locations,Reads per insertion location</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>filename</code></strong> :&ensp;<code>str</code></dt>
<dd>Path with the filename extension included(e.g "data_file/file.txt") describing where do you want to store the results.
By default it will be stored in the same location as the bamfile, with the same basename.
Example, if the bamfile path is data_file/data_1.bam then the per_gene file will be data_file/data_1.bam_pergene_insertions.txt</dd>
<dt><strong><code>tn_coordinates</code></strong> :&ensp;<code>dict</code></dt>
<dd>Last output of the function get_insertions <em>,</em>, tn_coordinates_per_gene = get_insertions_and_reads(
gene_coordinates, tncoordinatescopy_array, readnumb_array)</dd>
<dt><strong><code>gene_coordinates</code></strong> :&ensp;<code>dict </code></dt>
<dd>Output of the function add_chromosome_length:
gene_coordinates = add_chromosome_length(
gene_coordinates, chr_lengths_cumsum, ref_tid_roman)</dd>
<dt><strong><code>chr_lengths_cumsum</code></strong> :&ensp;<code>dict</code></dt>
<dd>last output of the function get_sequence_length : _, chr_lengths_cumsum = get_sequence_length(bam)</dd>
<dt><strong><code>ref_tid_roman</code></strong> :&ensp;<code>dict </code></dt>
<dd>Dictionary describing roman names as keys ,
ref_romannums = chromosomename_roman_to_arabic()[1]
ref_tid_roman = {key: value for key, value in zip(ref_romannums, ref_tid)}</dd>
<dt><strong><code>aliases_designation</code></strong> :&ensp;<code>dict</code></dt>
<dd>Last output of the function read_genes
<em>, </em>, aliases_designation = read_genes(
gff_file, essential_file, gene_name_file)</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def save_per_gene_insertions(
    filename,
    tn_coordinates,
    gene_coordinates,
    chr_lengths_cumsum,
    ref_tid_roman,
    aliases_designation,
):
    &#34;&#34;&#34;This function write in txt file 5 columns of information about transposon insertions separated by tabs. 
    The tabs are: Gene name,Chromosome,Start location,End location,Insertion locations,Reads per insertion location

    Parameters
    ----------
    filename : str
        Path with the filename extension included(e.g &#34;data_file/file.txt&#34;) describing where do you want to store the results.
        By default it will be stored in the same location as the bamfile, with the same basename. 
        Example, if the bamfile path is data_file/data_1.bam then the per_gene file will be data_file/data_1.bam_pergene_insertions.txt
    tn_coordinates : dict
         Last output of the function get_insertions _,_, tn_coordinates_per_gene = get_insertions_and_reads(
        gene_coordinates, tncoordinatescopy_array, readnumb_array)
    gene_coordinates : dict 
        Output of the function add_chromosome_length:  gene_coordinates = add_chromosome_length(
        gene_coordinates, chr_lengths_cumsum, ref_tid_roman)
    chr_lengths_cumsum : dict
        last output of the function get_sequence_length : _, chr_lengths_cumsum = get_sequence_length(bam)
    ref_tid_roman : dict 
        Dictionary describing roman names as keys , 
        ref_romannums = chromosomename_roman_to_arabic()[1]
        ref_tid_roman = {key: value for key, value in zip(ref_romannums, ref_tid)}
    aliases_designation : dict
        Last output of the function read_genes  _, _, aliases_designation = read_genes(
        gff_file, essential_file, gene_name_file)
    &#34;&#34;&#34;

    with open(filename, &#34;w&#34;) as f:

        f.write(
            &#34;Gene name\tChromosome\tStart location\tEnd location\tInsertion locations\tReads per insertion location\n&#34;
        )

        for gene in tn_coordinates:
            gene_chrom = ref_tid_roman.get(gene_coordinates.get(gene)[0])
            tncoordinates = [
                ins - chr_lengths_cumsum.get(gene_chrom)
                for ins in tn_coordinates[gene][3]
            ]

            if gene in aliases_designation:
                gene_alias = aliases_designation.get(gene)[0]
            else:
                gene_alias = gene

            f.write(
                gene_alias
                + &#34;\t&#34;
                + str(tn_coordinates[gene][0])
                + &#34;\t&#34;
                + str(tn_coordinates[gene][1] - chr_lengths_cumsum.get(gene_chrom))
                + &#34;\t&#34;
                + str(tn_coordinates[gene][2] - chr_lengths_cumsum.get(gene_chrom))
                + &#34;\t&#34;
                + str(tncoordinates)
                + &#34;\t&#34;
                + str(tn_coordinates[gene][4])
                + &#34;\n&#34;
            )</code></pre>
</details>
</dd>
</dl>
</section>
<section>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="transposonmapper" href="../index.html">transposonmapper</a></code></li>
</ul>
</li>
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="">
<li><code><a title="transposonmapper.exporting.save_as_bed" href="#transposonmapper.exporting.save_as_bed">save_as_bed</a></code></li>
<li><code><a title="transposonmapper.exporting.save_as_wig" href="#transposonmapper.exporting.save_as_wig">save_as_wig</a></code></li>
<li><code><a title="transposonmapper.exporting.save_per_essential_insertions" href="#transposonmapper.exporting.save_per_essential_insertions">save_per_essential_insertions</a></code></li>
<li><code><a title="transposonmapper.exporting.save_per_gene" href="#transposonmapper.exporting.save_per_gene">save_per_gene</a></code></li>
<li><code><a title="transposonmapper.exporting.save_per_gene_insertions" href="#transposonmapper.exporting.save_per_gene_insertions">save_per_gene_insertions</a></code></li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html>