<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>transposonmapper.mapping API documentation</title>
<meta name="description" content="" />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>transposonmapper.mapping</code></h1>
</header>
<section id="section-intro">
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">from .get_insertions_and_reads import get_insertions_and_reads
from .get_reads import get_reads
from .concatenate_chromosomes import (
    add_chromosome_length,
    add_chromosome_length_inserts,
)
from .find_chromosome_reads import find_chromosome_reads
from .correct_read_position import correct_read_position


__all__ = [
    &#34;get_insertions_and_reads&#34;,
    &#34;get_reads&#34;,
    &#34;add_chromosome_length&#34;,
    &#34;add_chromosome_length_inserts&#34;,
    &#34;find_chromosome_reads&#34;,
    &#34;correct_read_position&#34;
]</code></pre>
</details>
</section>
<section>
<h2 class="section-title" id="header-submodules">Sub-modules</h2>
<dl>
<dt><code class="name"><a title="transposonmapper.mapping.concatenate_chromosomes" href="concatenate_chromosomes.html">transposonmapper.mapping.concatenate_chromosomes</a></code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt><code class="name"><a title="transposonmapper.mapping.samflag" href="samflag.html">transposonmapper.mapping.samflag</a></code></dt>
<dd>
<div class="desc"><p>Created on Mon Jan
4 09:38:16 2021 â€¦</p></div>
</dd>
</dl>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="transposonmapper.mapping.add_chromosome_length"><code class="name flex">
<span>def <span class="ident">add_chromosome_length</span></span>(<span>coordinates, chr_lengths_cumsum, ref_tid_roman)</span>
</code></dt>
<dd>
<div class="desc"><p>This function returns a dictionary that for every gene , there is the chromosome number
information of where the gene belongs to , the coordinates for the start position, the end position
and the direction of the gene.
Parameters</p>
<hr>
<dl>
<dt><strong><code>coordinates</code></strong> :&ensp;<code>dict </code></dt>
<dd>This dictionary is the 1st output from the function read_genes(gff_file, essentials_file, gene_names_file)</dd>
<dt><strong><code>chr_lengths_cumsum</code></strong> :&ensp;<code>dict</code></dt>
<dd>This dictionary is the 2nd output from the function get_sequence_length(bam)</dd>
<dt><strong><code>ref_tid_roman</code></strong> :&ensp;<code>dict</code></dt>
<dd>This dictionary is the output from
ref_tid_roman = {key: value for key, value in zip(ref_romannums, ref_tid)}, where
ref_romannums = chromosomename_roman_to_arabic()[1] and
ref_tid = get_chromosome_names(bam)</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>coordinates</code></strong> :&ensp;<code>dict</code></dt>
<dd>A dictionary that for every gene , there is the chromosome number</dd>
<dt><code>information</code> of <code>where the gene belongs to , the coordinates for the start position, the end position</code></dt>
<dd>&nbsp;</dd>
</dl>
<p>and the direction of the gene.</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def add_chromosome_length(coordinates, chr_lengths_cumsum, ref_tid_roman):
    
    &#34;&#34;&#34;This function returns a dictionary that for every gene , there is the chromosome number
    information of where the gene belongs to , the coordinates for the start position, the end position
    and the direction of the gene. 
    Parameters
    ----------
    coordinates : dict 
        This dictionary is the 1st output from the function read_genes(gff_file, essentials_file, gene_names_file)
    chr_lengths_cumsum : dict
        This dictionary is the 2nd output from the function get_sequence_length(bam)
    ref_tid_roman : dict
        This dictionary is the output from 
        ref_tid_roman = {key: value for key, value in zip(ref_romannums, ref_tid)}, where
        ref_romannums = chromosomename_roman_to_arabic()[1] and 
        ref_tid = get_chromosome_names(bam)
        

    Returns
    -------
    coordinates: dict
        A dictionary that for every gene , there is the chromosome number
    information of where the gene belongs to , the coordinates for the start position, the end position
    and the direction of the gene. 
    &#34;&#34;&#34;  

    for key in coordinates:
        gene_chrom = ref_tid_roman.get(coordinates.get(key)[0])
        coordinates[key][1] = coordinates.get(key)[1] + chr_lengths_cumsum.get(gene_chrom)
        coordinates[key][2] = coordinates.get(key)[2] + chr_lengths_cumsum.get(gene_chrom)
    
    return coordinates</code></pre>
</details>
</dd>
<dt id="transposonmapper.mapping.add_chromosome_length_inserts"><code class="name flex">
<span>def <span class="ident">add_chromosome_length_inserts</span></span>(<span>coordinates, ref_names, chr_lengths)</span>
</code></dt>
<dd>
<div class="desc"><p>For each insertion location, add the length of all previous chromosomes</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>coordinates</code></strong> :&ensp;<code>numpy.array</code></dt>
<dd>readnumb_array, tncoordinates_array, coordinates = get_reads(bam)</dd>
<dt><strong><code>ref_names</code></strong> :&ensp;<code>list</code></dt>
<dd>ref_tid = get_chromosome_names(bam)
ref_names = list(ref_tid.keys())</dd>
<dt><strong><code>chr_lengths</code></strong> :&ensp;<code>dict</code></dt>
<dd>chr_lengths, _ = get_sequence_length(bam)</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>coordinates</code></strong> :&ensp;<code>numpy.array</code></dt>
<dd>&nbsp;</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def add_chromosome_length_inserts(coordinates, ref_names, chr_lengths):
    &#34;&#34;&#34; For each insertion location, add the length of all previous chromosomes

    Parameters
    ----------
    coordinates : numpy.array
        readnumb_array, tncoordinates_array, coordinates = get_reads(bam)
    ref_names : list
        ref_tid = get_chromosome_names(bam)
        ref_names = list(ref_tid.keys())
    chr_lengths : dict
        chr_lengths, _ = get_sequence_length(bam)
   

    Returns
    -------
    coordinates: numpy.array
        
    &#34;&#34;&#34;
    
    ll = 0
    for ii in range(1,len(ref_names)):
        ll += chr_lengths[ref_names[ii-1]]
        aa = np.where(coordinates[:,0] == ii + 1)
        coordinates[aa,1] = coordinates[aa,1] + ll

    return coordinates</code></pre>
</details>
</dd>
<dt id="transposonmapper.mapping.correct_read_position"><code class="name flex">
<span>def <span class="ident">correct_read_position</span></span>(<span>flags, start_position, readlength)</span>
</code></dt>
<dd>
<div class="desc"><p>Correct starting position for reads with reversed orientation</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def correct_read_position(flags, start_position, readlength):
    &#34;&#34;&#34;
    Correct starting position for reads with reversed orientation    

    &#34;&#34;&#34;

    flag0coor_array = np.where(flags == 1)  # coordinates reads 5&#39; -&gt; 3&#39;
    flag16coor_array = np.where(flags == -1)  # coordinates reads 3&#39; -&gt; 5&#39;

    startdirect_array = start_position[flag0coor_array]
    flagdirect_array = flags[flag0coor_array]

    startindirect_array = (
        start_position[flag16coor_array] + readlength[flag16coor_array]
    )
    flagindirect_array = flags[flag16coor_array]

    start_position_corrected = np.concatenate(
        (startdirect_array, startindirect_array), axis=0
    )
    flags_corrected = np.concatenate((flagdirect_array, flagindirect_array), axis=0)

    start_sortindices = start_position_corrected.argsort(
        kind=&#34;mergesort&#34;
    )  # use mergesort for stable sorting
    start_position_corrected = start_position_corrected[start_sortindices]
    flags_corrected = flags_corrected[start_sortindices]

    return start_position_corrected, flags_corrected</code></pre>
</details>
</dd>
<dt id="transposonmapper.mapping.find_chromosome_reads"><code class="name flex">
<span>def <span class="ident">find_chromosome_reads</span></span>(<span>chromosome, N_reads:Â int)</span>
</code></dt>
<dd>
<div class="desc"><p>Syntax: [flags, start_position, readlength] = get_chromosome_reads(chromosome, N_reads)</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def find_chromosome_reads(chromosome, N_reads: int):
    &#34;&#34;&#34;
    
    Syntax: [flags, start_position, readlength] = get_chromosome_reads(chromosome, N_reads)


    &#34;&#34;&#34;
    # Initialize arrays
    start_position = np.empty(shape=(N_reads), dtype=int)
    flags = np.empty(shape=(N_reads), dtype=int)
    readlength = np.empty(shape=(N_reads), dtype=int)

    for index, reads in enumerate(chromosome):
        read = str(reads).split(&#34;\t&#34;)

        start_position[index] = int(read[3]) + 1

        # GET FLAG FOR EACH READ. IF READ ON FORWARD STRAND, ASSIGN VALUE 1, IF READ ON REVERSE STRAND ASSIGN VALUE -1, IF READ UNMAPPED OR SECONDARY ALIGNMENT ASSIGN VALUE 0
        #            flag_array[read_counter] = int(read[1])
        samprop = samflags(flag=int(read[1]), verbose=False)[1]
        if &#34;read reverse strand&#34; in samprop:
            flags[index] = -1
        else:
            flags[index] = 1
        if &#34;not primary alignment&#34; in samprop or &#34;read unmapped&#34; in samprop:
            flags[index] = 0

        cigarmatch_list = []
        if not reads.cigartuples == None:
            for cigar_type, cigar_length in reads.cigartuples:
                if cigar_type == 0:
                    cigarmatch_list.append(cigar_length)
                elif cigar_type == 2:
                    cigarmatch_list.append(cigar_length)
        match_length = sum(cigarmatch_list)

        readlength[index] = match_length  # int(len(read[9]))

    return flags, start_position, readlength</code></pre>
</details>
</dd>
<dt id="transposonmapper.mapping.get_insertions_and_reads"><code class="name flex">
<span>def <span class="ident">get_insertions_and_reads</span></span>(<span>coordinates, tn_coordinates, readnumb_array)</span>
</code></dt>
<dd>
<div class="desc"><p>This function computes the total number of transposons per gene , the number of reads per gene and
the distribution
of transposons along the gene.</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>coordinates</code></strong> :&ensp;<code>dict </code></dt>
<dd>This is the output of the function add_chromosome_length(coordinates, chr_lengths_cumsum, ref_tid_roman)</dd>
<dt><strong><code>tn_coordinates</code></strong> :&ensp;<code>numpy.array</code></dt>
<dd>This is the output of the function add_chromosome_length_inserts(coordinates, ref_names, chr_lengths)</dd>
<dt><strong><code>readnumb_array</code></strong> :&ensp;<code>numpy.array</code></dt>
<dd>This is the 1st output of the function get_reads(bam)</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>tn_per_gene</code></strong> :&ensp;<code>dict </code></dt>
<dd>A dict which every key corresponds with each gene and each value with the total number of transposons found in that gene</dd>
<dt><strong><code>reads_per_gene</code></strong> :&ensp;<code>dict</code></dt>
<dd>A dict which every key corresponds with each gene and each value with the total number of reads for all the transposons found in that gene</dd>
<dt><strong><code>tn_coordinates_per_gene</code></strong> :&ensp;<code>dict </code></dt>
<dd>A dict which every key corresponds with each gene and each value with a
list of 4 elements:
- the chromosome number
- gene start position
- gene end position
- distribution of reads per transposon found inside the gene</dd>
</dl></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def get_insertions_and_reads(coordinates, tn_coordinates, readnumb_array):
    &#34;&#34;&#34;This function computes the total number of transposons per gene , the number of reads per gene and
    the distribution  of transposons along the gene.

    Parameters
    ----------
    coordinates : dict 
        This is the output of the function add_chromosome_length(coordinates, chr_lengths_cumsum, ref_tid_roman)
    tn_coordinates : numpy.array
        This is the output of the function add_chromosome_length_inserts(coordinates, ref_names, chr_lengths)
    readnumb_array : numpy.array
        This is the 1st output of the function get_reads(bam)

    Returns
    -------
    tn_per_gene : dict 
        A dict which every key corresponds with each gene and each value with the total number of transposons found in that gene
    reads_per_gene: dict
        A dict which every key corresponds with each gene and each value with the total number of reads for all the transposons found in that gene

    tn_coordinates_per_gene: dict 
         A dict which every key corresponds with each gene and each value with a  list of 4 elements:
            - the chromosome number
            - gene start position
            - gene end position
            - distribution of reads per transposon found inside the gene 

    &#34;&#34;&#34;

    tn_per_gene = {}
    reads_per_gene = {}
    tn_coordinates_per_gene = {}

    for gene in coordinates:
        xx = np.where(np.logical_and(tn_coordinates[:,1] &gt;= coordinates.get(gene)[1], tn_coordinates[:,1] &lt;= coordinates.get(gene)[2])) #get all insertions within range of current gene
        tn_per_gene[gene] = np.size(xx)
        reads_per_gene[gene] = sum(readnumb_array[xx]) - max(readnumb_array[xx], default=0) #REMOVE LARGEST VALUE TO REDUCE NOISE

        if np.size(xx) &gt; 0:
            tn_coordinates_per_gene[gene] = [coordinates.get(gene)[0], coordinates.get(gene)[1], coordinates.get(gene)[2], list(tn_coordinates[xx[0][0]:xx[0][-1]+1, 1]), list(readnumb_array[xx])]
        else:
            tn_coordinates_per_gene[gene] = [coordinates.get(gene)[0], coordinates.get(gene)[1], coordinates.get(gene)[2], [], []]

    return tn_per_gene, reads_per_gene, tn_coordinates_per_gene</code></pre>
</details>
</dd>
<dt id="transposonmapper.mapping.get_reads"><code class="name flex">
<span>def <span class="ident">get_reads</span></span>(<span>bam)</span>
</code></dt>
<dd>
<div class="desc"><p>This function retrieves
all reads within a specified genomic region. </p>
<p>Usage</p>
<hr>
<p>file1,file2,file3=get_reads(bam)</p>
<p>Parameters</p>
<hr>
<p>bam :
The output for the function pysam.AlignmentFile(bamfile, "rb")</p>
<p>Returns</p>
<hr>
<p>readnumb_array: reads per genomic region
tncoordinates_array: Array of three columns where the 2nd one indicated the start position where there was a transposon
tncoordinatescopy_array: A copy from tncoordinates_array</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def get_reads(bam):
    &#34;&#34;&#34;
 This function retrieves  all reads within a specified genomic region. 
     
    Usage
    ----------
    file1,file2,file3=get_reads(bam)

    Parameters
    ----------
    bam :  The output for the function pysam.AlignmentFile(bamfile, &#34;rb&#34;)

    Returns
    -------
    readnumb_array: reads per genomic region
    tncoordinates_array: Array of three columns where the 2nd one indicated the start position where there was a transposon
    tncoordinatescopy_array: A copy from tncoordinates_array
    &#34;&#34;&#34;

    # Get chromosome properties
    ref_tid = get_chromosome_names(bam)
    ref_names = list(ref_tid.keys())
    chr_lengths, _ = get_sequence_length(bam)
    chr_mapped_reads = get_chromosome_reads(bam)

    ### GET ALL READS WITHIN A SPECIFIED GENOMIC REGION
    tnnumber_dict = {}
    unique_insertions = 0

    for chr_num in ref_names:
        timer_start = timeit.default_timer()

        print(&#34;Getting reads for chromosome %s ...&#34; % chr_num)
        chromosome = bam.fetch(chr_num, 0, chr_lengths[chr_num], until_eof=True)
        N_reads = chr_mapped_reads[chr_num][2]

        [flags, start_position, readlength] = find_chromosome_reads(chromosome, N_reads)

        start_position_corrected, flags_corrected = correct_read_position(
            flags, start_position, readlength
        )

        # CREATE ARRAY OF START POSITION AND FLAGS OF ALL READS IN GENOME
        ref_tid_kk = int(ref_tid[chr_num] + 1)
        if unique_insertions == 0:
            tncoordinates_array = np.array([])

        unique_insertions_per_read = 0  # Number of unique reads per insertion
        num_transposons = 1  # Number of unique reads in current chromosome (Number of transposons in current chromosome)
        for ii in range(1, len(start_position_corrected)):
            if (
                abs(start_position_corrected[ii] - start_position_corrected[ii - 1])
                &lt;= 2
                and flags_corrected[ii] == flags_corrected[ii - 1]
            ):  # If two subsequent reads are within two basepairs and have the same orientation, add them together.
                unique_insertions_per_read += 1
            else:
                avg_start_pos = abs(
                    round(
                        np.mean(
                            start_position_corrected[
                                ii - unique_insertions_per_read - 1 : ii
                            ]
                        )
                    )
                )
                if tncoordinates_array.size == 0:  # include first read
                    tncoordinates_array = np.array(
                        [ref_tid_kk, int(avg_start_pos), int(flags_corrected[ii - 1])]
                    )
                    readnumb_list = [unique_insertions_per_read + 1]
                else:
                    tncoordinates_array = np.vstack(
                        (
                            tncoordinates_array,
                            [
                                ref_tid_kk,
                                int(avg_start_pos),
                                int(flags_corrected[ii - 1]),
                            ],
                        )
                    )
                    readnumb_list.append(unique_insertions_per_read + 1)
                unique_insertions_per_read = 0
                num_transposons += 1
                unique_insertions += 1

            if ii == len(start_position_corrected) - 1:  # include last read
                avg_start_pos = abs(
                    round(
                        np.mean(
                            start_position_corrected[
                                ii - unique_insertions_per_read - 1 : ii
                            ]
                        )
                    )
                )
                tncoordinates_array = np.vstack(
                    (
                        tncoordinates_array,
                        [ref_tid_kk, int(avg_start_pos), int(flags_corrected[ii - 1])],
                    )
                )
                readnumb_list.append(unique_insertions_per_read + 1)

        tnnumber_dict[chromosome] = num_transposons

        timer_end = timeit.default_timer()
        print(
            &#34;Chromosome %s completed in %.3f seconds&#34;
            % (chromosome, (timer_end - timer_start))
        )
        print(&#34;&#34;)

    readnumb_array = np.array(readnumb_list)

    tncoordinatescopy_array = np.array(tncoordinates_array, copy=True)

    return readnumb_array, tncoordinates_array, tncoordinatescopy_array</code></pre>
</details>
</dd>
</dl>
</section>
<section>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="transposonmapper" href="../index.html">transposonmapper</a></code></li>
</ul>
</li>
<li><h3><a href="#header-submodules">Sub-modules</a></h3>
<ul>
<li><code><a title="transposonmapper.mapping.concatenate_chromosomes" href="concatenate_chromosomes.html">transposonmapper.mapping.concatenate_chromosomes</a></code></li>
<li><code><a title="transposonmapper.mapping.samflag" href="samflag.html">transposonmapper.mapping.samflag</a></code></li>
</ul>
</li>
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="">
<li><code><a title="transposonmapper.mapping.add_chromosome_length" href="#transposonmapper.mapping.add_chromosome_length">add_chromosome_length</a></code></li>
<li><code><a title="transposonmapper.mapping.add_chromosome_length_inserts" href="#transposonmapper.mapping.add_chromosome_length_inserts">add_chromosome_length_inserts</a></code></li>
<li><code><a title="transposonmapper.mapping.correct_read_position" href="#transposonmapper.mapping.correct_read_position">correct_read_position</a></code></li>
<li><code><a title="transposonmapper.mapping.find_chromosome_reads" href="#transposonmapper.mapping.find_chromosome_reads">find_chromosome_reads</a></code></li>
<li><code><a title="transposonmapper.mapping.get_insertions_and_reads" href="#transposonmapper.mapping.get_insertions_and_reads">get_insertions_and_reads</a></code></li>
<li><code><a title="transposonmapper.mapping.get_reads" href="#transposonmapper.mapping.get_reads">get_reads</a></code></li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html>