
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Parameters &#8212; SATAY pipeline at Delft :)</title>
    
  <link href="../../_static/css/theme.css" rel="stylesheet" />
  <link href="../../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/sphinx-book-theme.acff12b8f9c144ce68a297486a2fa670.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/tabs.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/togglebutton.js"></script>
    <script src="../../_static/clipboard.min.js"></script>
    <script src="../../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../_static/sphinx-thebe.js"></script>
    <link rel="shortcut icon" href="../../_static/genetica_modified_neon.png"/>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="&lt;no title&gt;" href="verify_data_files.html" />
    <link rel="prev" title="Transposonmapper output data postprocessing" href="../../06-Tutorials/tutorial.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../../index.html">
      
      <img src="../../_static/genetica_modified_neon.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">SATAY pipeline at Delft :)</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../Introduction.html">
   Status of the project
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Getting started with transposonmapper
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../getting-started/00-getting-started.html">
   Installation
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  What transposonmapper does?
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../00-Intro/flowchart.html">
   Flowchart of the pipeline
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../00-Intro/data-analysis-workflow.html">
   Overview of the pipeline functions
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../00-Intro/satay_analysis_notes.html">
   What is SATAY
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Processing workflow
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../01-bash-GUI/00-guidelines.html">
   How to use the pipeline in the docker container
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../01-bash-GUI/01-processing_workflow.html">
   Code for the bash GUI
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Docker
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../03-docker-doc/00-Docker-Users.html">
   Build your own docker container with ours satay pipeline
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Contributing guidelines
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../04-Developers/00-contributing-guidelines.html">
   Contributing guidelines for developers
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Code of conduct
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../05-Code-of-conduct/00-code-of-conduct.html">
   Code of Conduct
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tutorials
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../06-Tutorials/tutorial.html">
   Transposonmapper output data postprocessing
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  API reference
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Parameters
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../_sources/api_reference/importing/index.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/SATAY-LL/SATAY-jupyter-book"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/SATAY-LL/SATAY-jupyter-book/issues/new?title=Issue%20on%20page%20%2Fapi_reference/importing/index.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <p>&lt;!doctype html&gt;</p>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>transposonmapper.importing API documentation</title>
<meta name="description" content="" />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>transposonmapper.importing</code></h1>
</header>
<section id="section-intro">
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">from .verify_data_files import verify_data_files
from .load_default_files import load_default_files,load_sgd_tab
from .read_genes import read_genes
<p><strong>all</strong> = [
“load_default_files”,
“read_genes”,
“load_sgd_tab”
]</code></pre></p>
</details>
</section>
<section>
<h2 class="section-title" id="header-submodules">Sub-modules</h2>
<dl>
<dt><code class="name"><a title="transposonmapper.importing.verify_data_files" href="verify_data_files.html">transposonmapper.importing.verify_data_files</a></code></dt>
<dd>
<div class="desc"></div>
</dd>
</dl>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="transposonmapper.importing.load_default_files"><code class="name flex">
<span>def <span class="ident">load_default_files</span></span>(<span>gff_file=None, essentials_file=None, gene_names_file=None)</span>
</code></dt>
<dd>
<div class="desc"><p>This function loads some files that have a recurrent use throughout the pipeline.
It will look inside the satay/data_files folder for the files if the input is None.
Otherwise it will return the same input file.
</p>
<pre><code>Usage:
------
file1, file2, file3=load_default_files(gff_file=file1,
                                        essentials_file=file2,
                                        gene_names_file=file3)
if you dont provide a specific file as an input , then it will look into the default folder
satay/data_files/ and it will give the standard file provided in the package.
<p>file1, file2, file3=load_default_files(gff_file=None,
essentials_file=None,
gene_names_file=None)</p>
<div class="section" id="parameters">
<h1>Parameters<a class="headerlink" href="#parameters" title="Permalink to this headline">¶</a></h1>
<p>gff_file : [.gff3] –&gt; Annotated genome from Saccharomyces cerevisiae (baker’s yeast)
(&lt;<a class="reference external" href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.gff.gz">https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.gff.gz</a>&gt;)
essentials_file : [.txt] –&gt;  Essentials genes annotated from yeast , written using the systematic name standard ,
all in one column
gene_names_file : [.txt] –&gt; This documents lists all the Saccharomyces cerevisiae S288c entries present in
</code></pre></p>
<p>this release of UniProtKB/Swiss-Prot. Yeast (Saccharomyces cerevisiae): entries, gene names and cross-references to SGD.
Release:
2021_01 of 10-Feb-2021</p>
<pre><code>Returns
-------
gff_file : [.gff3]--&gt; Annotated genome from Saccharomyces cerevisiae (baker's yeast) 
    (&lt;https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.gff.gz&gt;)
essentials_file : [.txt]--&gt;  Essentials genes annotated from yeast , written using the systematic name standard , 
    all in one column
gene_names_file : [.txt]--&gt; This documents lists all the Saccharomyces cerevisiae S288c entries present in
</code></pre>
<p>this release of UniProtKB/Swiss-Prot. Yeast (Saccharomyces cerevisiae): entries, gene names and cross-references to SGD.
Release:
2021_01 of 10-Feb-2021</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def load_default_files(gff_file=None, essentials_file=None, gene_names_file=None):
    &#34;&#34;&#34;This function loads some files that have a recurrent use throughout the pipeline.
    It will look inside the satay/data_files folder for the files if the input is None. 
    Otherwise it will return the same input file.  
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>Usage:
------
file1, file2, file3=load_default_files(gff_file=file1,
                                        essentials_file=file2,
                                        gene_names_file=file3)
if you dont provide a specific file as an input , then it will look into the default folder
satay/data_files/ and it will give the standard file provided in the package.

file1, file2, file3=load_default_files(gff_file=None,
                                        essentials_file=None,
                                        gene_names_file=None)

Parameters
----------
gff_file : [.gff3] --&amp;gt; Annotated genome from Saccharomyces cerevisiae (baker&amp;#39;s yeast) 
    (https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.gff.gz)
essentials_file : [.txt] --&amp;gt;  Essentials genes annotated from yeast , written using the systematic name standard , 
    all in one column
gene_names_file : [.txt] --&amp;gt; This documents lists all the Saccharomyces cerevisiae S288c entries present in
</pre></div>
</div>
<p>this release of UniProtKB/Swiss-Prot. Yeast (Saccharomyces cerevisiae): entries, gene names and cross-references to SGD.
Release:     2021_01 of 10-Feb-2021</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>Returns
-------
gff_file : [.gff3]--&amp;gt; Annotated genome from Saccharomyces cerevisiae (baker&amp;#39;s yeast) 
    (https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.gff.gz)
essentials_file : [.txt]--&amp;gt;  Essentials genes annotated from yeast , written using the systematic name standard , 
    all in one column
gene_names_file : [.txt]--&amp;gt; This documents lists all the Saccharomyces cerevisiae S288c entries present in
</pre></div>
</div>
<p>this release of UniProtKB/Swiss-Prot. Yeast (Saccharomyces cerevisiae): entries, gene names and cross-references to SGD.
Release:     2021_01 of 10-Feb-2021
“””</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>default_path = pkg_resources.resource_filename(&amp;#34;transposonmapper&amp;#34;, &amp;#34;data_files/&amp;#34;)
if gff_file is None:
    gff_file = os.path.join(
        default_path, &amp;#34;Saccharomyces_cerevisiae.R64-1-1.99.gff3&amp;#34;
    )
if essentials_file is None:
    essentials_file = os.path.join(
        default_path, &amp;#34;Cerevisiae_AllEssentialGenes_List.txt&amp;#34;
    )
if gene_names_file is None:
    gene_names_file = os.path.join(default_path, &amp;#34;Yeast_Protein_Names.txt&amp;#34;)


return gff_file, essentials_file, gene_names_file&lt;/code&gt;&lt;/pre&gt;
</pre></div>
</div>
</details>
</dd>
<dt id="transposonmapper.importing.load_sgd_tab"><code class="name flex">
<span>def <span class="ident">load_sgd_tab</span></span>(<span>sgd_features_file=None)</span>
</code></dt>
<dd>
<div class="desc"><p>This function loads the file SGD_features.tab
The latest version of the SGD_features.tab file is based on Genome Version R64-2-1.
If a specific file is provided it will output that file , otherwise , if it is set to None then
it will give the standard file provided in the package.
Usage:
file1=load_sgd_tab(sgd_features_file=file1)
if you dont provide a specific file as an input , then it will look into the default folder
satay/data_files/ and it will give the standard file provided in the package.</p>
<p>file1=load_sgd_tab(sgd_features_file=None)</p>
<h2 id="arguments">Arguments</h2>
<p>sgd_features_file {.gff3} &ndash; The latest version of the SGD_features.tab file is based on Genome Version R64-2-1.</p>
<h2 id="returns">Returns</h2>
<p>sgd_features_file {.gff3} &ndash; The latest version of the SGD_features.tab file is based on Genome Version R64-2-1.</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def load_sgd_tab(sgd_features_file=None):
    &#34;&#34;&#34;This function loads the file SGD_features.tab
    The latest version of the SGD_features.tab file is based on Genome Version R64-2-1.
    If a specific file is provided it will output that file , otherwise , if it is set to None then 
    it will give the standard file provided in the package. 
    Usage:
    file1=load_sgd_tab(sgd_features_file=file1)
    if you dont provide a specific file as an input , then it will look into the default folder
    satay/data_files/ and it will give the standard file provided in the package.
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>file1=load_sgd_tab(sgd_features_file=None)


Arguments:
    sgd_features_file {.gff3} -- The latest version of the SGD_features.tab file is based on Genome Version R64-2-1.

Returns:
    sgd_features_file {.gff3} -- The latest version of the SGD_features.tab file is based on Genome Version R64-2-1.
&amp;#34;&amp;#34;&amp;#34;
default_path = pkg_resources.resource_filename(&amp;#34;transposonmapper&amp;#34;, &amp;#34;data_files/&amp;#34;)
if sgd_features_file is None: 

    sgd_features_file = os.path.join(default_path,&amp;#39;SGD_features.tab&amp;#39;)

return sgd_features_file&lt;/code&gt;&lt;/pre&gt;
</pre></div>
</div>
</details>
</dd>
<dt id="transposonmapper.importing.read_genes"><code class="name flex">
<span>def <span class="ident">read_genes</span></span>(<span>gff_file, essentials_file, gene_names_file)</span>
</code></dt>
<dd>
<div class="desc"><p>This function reads the useful information inside the gff_file, essentials_file and
gene_names_file.
For the gff_file and essentials_file extracts the gene coordinates , specifying the chromosome, start ,end
and direction.
For the gene_names_files it translates the systematic name into the standard name.</p>
<pre><code>Usage:
<p>[gene_coordinates, essential_coordinates, aliases_designation] =
read_genes(gff_file, essentials_file, gene_names_file)</p>
<p>Inputs:
- gff_file : Annotated genome from Saccharomyces cerevisiae (baker’s yeast) (&lt;<a class="reference external" href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.gff.gz">https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.gff.gz</a>&gt;)
- essentials_file:  Essentials genes annotated from yeast , written using the systematic name standard , all in one column
- gene_names_file:  This documents lists all the Saccharomyces cerevisiae S288c entries present in
</code></pre></p>
<p>this release of UniProtKB/Swiss-Prot.
Yeast (Saccharomyces cerevisiae): entries, gene names and cross-references to SGD.
Release:
2021_01 of 10-Feb-2021</p>
<pre><code>Outputs:
    gene_coordinates : a dict specifying for each gene the chromosome number the gene
    belongs to, the start gene coordinate, the end gene coordinate and the strand direction ('+' or '-'). 
    essential_coordinates: a dict specifying for each annotated essential gene the chromosome number the gene
    belongs to, the start gene coordinate, the end gene coordinate and the strand direction ('+' or '-'). 
    aliases_designation: a dict that for each systematic gene name specify the standard gene name.
</code></pre></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def read_genes(gff_file, essentials_file, gene_names_file):
    &#34;&#34;&#34;
    This function reads the useful information inside the gff_file, essentials_file and
    gene_names_file. 
    For the gff_file and essentials_file extracts the gene coordinates , specifying the chromosome, start ,end 
    and direction. 
    For the gene_names_files it translates the systematic name into the standard name.
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>Usage: 

 [gene_coordinates, essential_coordinates, aliases_designation] = 
        read_genes(gff_file, essentials_file, gene_names_file)

Inputs:
    - gff_file : Annotated genome from Saccharomyces cerevisiae (baker&amp;#39;s yeast) (https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_genomic.gff.gz)
    - essentials_file:  Essentials genes annotated from yeast , written using the systematic name standard , all in one column
    - gene_names_file:  This documents lists all the Saccharomyces cerevisiae S288c entries present in
</pre></div>
</div>
<p>this release of UniProtKB/Swiss-Prot.
Yeast (Saccharomyces cerevisiae): entries, gene names and cross-references to SGD.
Release:     2021_01 of 10-Feb-2021</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>Outputs:
    gene_coordinates : a dict specifying for each gene the chromosome number the gene
    belongs to, the start gene coordinate, the end gene coordinate and the strand direction (&amp;#39;+&amp;#39; or &amp;#39;-&amp;#39;). 
    essential_coordinates: a dict specifying for each annotated essential gene the chromosome number the gene
    belongs to, the start gene coordinate, the end gene coordinate and the strand direction (&amp;#39;+&amp;#39; or &amp;#39;-&amp;#39;). 
    aliases_designation: a dict that for each systematic gene name specify the standard gene name.  

&amp;#34;&amp;#34;&amp;#34;

# Get gene position
gene_coordinates = gene_position(gff_file)  #&amp;#39;YAL069W&amp;#39; | [&amp;#39;I&amp;#39;, 335, 649], ...

# Get all annotated essential genes
essential_coordinates = {}
with open(essentials_file, &amp;#34;r&amp;#34;) as f:
    genes = f.readlines()[1:]
    for gene in genes:
        name = gene.strip(&amp;#34;\n&amp;#34;)
        essential_coordinates[name] = gene_coordinates.get(name).copy()

# Get aliases of all genes
aliases_designation = gene_aliases(gene_names_file)[0]  #&amp;#39;YMR056C&amp;#39; \ [&amp;#39;AAC1&amp;#39;], ...

return gene_coordinates, essential_coordinates, aliases_designation&lt;/code&gt;&lt;/pre&gt;
</pre></div>
</div>
</details>
</dd>
</dl>
</section>
<section>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="transposonmapper" href="../index.html">transposonmapper</a></code></li>
</ul>
</li>
<li><h3><a href="#header-submodules">Sub-modules</a></h3>
<ul>
<li><code><a title="transposonmapper.importing.verify_data_files" href="verify_data_files.html">transposonmapper.importing.verify_data_files</a></code></li>
</ul>
</li>
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="">
<li><code><a title="transposonmapper.importing.load_default_files" href="#transposonmapper.importing.load_default_files">load_default_files</a></code></li>
<li><code><a title="transposonmapper.importing.load_sgd_tab" href="#transposonmapper.importing.load_sgd_tab">load_sgd_tab</a></code></li>
<li><code><a title="transposonmapper.importing.read_genes" href="#transposonmapper.importing.read_genes">read_genes</a></code></li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html></div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./api_reference/importing"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="../../06-Tutorials/tutorial.html" title="previous page">Transposonmapper output data postprocessing</a>
    <a class='right-next' id="next-link" href="verify_data_files.html" title="next page">&lt;no title&gt;</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Gregory van Beek , Maurits Kok, Leila Iñigo de la Cruz<br/>
        
            &copy; Copyright Apache 2.0.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>